<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Johnson" />

<meta name="date" content="2022-09-28" />

<title>QC &amp; normalisation - sequencing data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">anderson-peffers_OA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="02_EDA.html">Exploratory data analysis</a>
    </li>
    <li>
      <a href="03_Differential_expression.html">Differential expression analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Small RNA sequencing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_Sequencing_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="05_Sequencing_EDA.html">Exploratory data analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Multi-omics analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06_jDR.html">Joint dimensionality reduction</a>
    </li>
    <li>
      <a href="07_Clustering.html">Clustering</a>
    </li>
    <li>
      <a href="08_Functional_enrichment.html">Functional enrichment</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ejjohnson93/anderson-peffers_OA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">QC &amp; normalisation - sequencing
data</h1>
<h4 class="author">Emily Johnson</h4>
<h4 class="date">2022-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>anderson-peffers_OA/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220927code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220927code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220927)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreeb8abb7ebb24e6e1d60f8a8c601795873111a34e1targetblankb8abb7ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/b8abb7ebb24e6e1d60f8a8c601795873111a34e1" target="_blank">b8abb7e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreeb8abb7ebb24e6e1d60f8a8c601795873111a34e1targetblankb8abb7ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/b8abb7ebb24e6e1d60f8a8c601795873111a34e1" target="_blank">b8abb7e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Unstaged changes:
    Modified:   analysis/03_Differential_expression.Rmd
    Modified:   analysis/04_Sequencing_QC_normalisation.Rmd
    Modified:   analysis/06_jDR.Rmd
    Modified:   analysis/07_Clustering.Rmd
    Modified:   analysis/08_Functional_enrichment.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd
    Modified:   code/utility-functions.R
    Modified:   data/plasma_lncRNA_data.rds
    Modified:   data/sf_lncRNA_data.rds
    Modified:   output/plasma_limma_DE_results_log2.xlsx
    Modified:   output/plasma_proDA_DE_results.xlsx
    Modified:   output/sf_proDA_DE_results.xlsx
    Modified:   plasma_model.hdf5
    Modified:   sf_model.hdf5

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/04_Sequencing_QC_normalisation.Rmd</code>) and HTML
(<code>docs/04_Sequencing_QC_normalisation.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/928fd819c7a6726f3ea0bc9bedec6e1910242471/analysis/04_Sequencing_QC_normalisation.Rmd" target="_blank">928fd81</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-08
</td>
<td>
jDR + EDA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/928fd819c7a6726f3ea0bc9bedec6e1910242471/docs/04_Sequencing_QC_normalisation.html" target="_blank">928fd81</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-08
</td>
<td>
jDR + EDA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/ae6f5f4c77d2458fec583f75b216e34d5a9478ad/docs/04_Sequencing_QC_normalisation.html" target="_blank">ae6f5f4</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-04
</td>
<td>
Knitted docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/dfeca72852e1aa1a9379ce9bf1c65108b4fec240/analysis/04_Sequencing_QC_normalisation.Rmd" target="_blank">dfeca72</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-04
</td>
<td>
Sequencing data processing
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/00c9ca773d66e9a333588cda0f77eb5d9e78346e/analysis/04_Sequencing_QC_normalisation.Rmd" target="_blank">00c9ca7</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-04
</td>
<td>
Helper functions norm etc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/00c9ca773d66e9a333588cda0f77eb5d9e78346e/docs/04_Sequencing_QC_normalisation.html" target="_blank">00c9ca7</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-04
</td>
<td>
Helper functions norm etc
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/e34c9f9d885837b14673c528fea1cce3b9824a07/analysis/04_Sequencing_QC_normalisation.Rmd" target="_blank">e34c9f9</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-03
</td>
<td>
Updated theme + additional analysis docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/c63c64ff60364617182a11660b1352d65c910cac/analysis/04_Sequencing_QC_normalisation.Rmd" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/c63c64ff60364617182a11660b1352d65c910cac/docs/04_Sequencing_QC_normalisation.html" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>source(&quot;./code/utility-functions.R&quot;) # load helper functions 
library(tidyverse)
library(edgeR)
library(cowplot) # ggplot2 theme
library(viridis) # colour schemes
library(ggpubr) # utilities to arrange/annotate ggplot2 plots
library(R.utils) # function to count zero values
library(PCAtools) # pca &amp; correlation plot</code></pre>
</div>
<div id="phenotype" class="section level1">
<h1>Phenotype</h1>
<p>First load the phenotype and sequencing info data:</p>
<pre class="r"><code># Pheno
plasma_pheno &lt;- read.csv(&quot;./data/plasma_pheno.csv&quot;)
sf_pheno &lt;- read.csv(&quot;./data/sf_pheno.csv&quot;)


# Sequencing info
plasma_sequencing_info &lt;- read.csv(&quot;./data/plasma_sequencing_info.csv&quot;)
sf_sequencing_info &lt;- read.csv(&quot;./data/sf_sequencing_info.csv&quot;)</code></pre>
<p>Process the phenotype and sequencing info tables and combine
them:</p>
<pre class="r"><code># Match pheno data to sequencing info
plasma_pheno &lt;- plasma_pheno[match(plasma_sequencing_info$customerID, plasma_pheno$Sample.Number),]
sf_pheno &lt;- sf_pheno[match(sf_sequencing_info$customerID, sf_pheno$Sample.Number),]


# Join sequencing and pheno information
plasma_pheno &lt;- cbind(plasma_pheno, plasma_sequencing_info)
sf_pheno &lt;- cbind(sf_pheno, sf_sequencing_info)


# Process phenotype data to include a &#39;sample_&#39; prefix in front of the sample IDs
# Convert horse ID to a factor
plasma_pheno$Sample.Number &lt;- paste0(&quot;Sample_&quot;, plasma_pheno$Sample.Number)
plasma_pheno$Horse &lt;- as.factor(plasma_pheno$Horse)

sf_pheno$Sample.Number &lt;- paste0(&quot;Sample_&quot;, sf_pheno$Sample.Number)
sf_pheno$Horse &lt;- as.factor(sf_pheno$Horse)
sf_pheno &lt;- sf_pheno[,-8]</code></pre>
<p>Next, load and process each type of sequencing data.</p>
</div>
<div id="mirna" class="section level1">
<h1>miRNA</h1>
<p>The data was provided in CPM format. The reports TAmiRNA provided
contain the raw counts for the miRNA. To assess the normalisation the
raw counts and the CPM normalised counts were both read in.</p>
<div id="plasma" class="section level2">
<h2>Plasma</h2>
<div id="read-data-in" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># Raw miRNA counts
plasma_miRNA_data_raw &lt;- read.csv(&quot;./data/raw_miRNA_plasma.csv&quot;, row.names = 1) 
colnames(plasma_miRNA_data_raw) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(plasma_miRNA_data_raw))


# CPM normalised plasma miRNA data
plasma_miRNA_data &lt;- read.csv(&quot;./data/miRNA_plasma.csv&quot;, row.names = 1)
colnames(plasma_miRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(plasma_miRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>plasma_miRNA_data &lt;- plasma_miRNA_data[,match(plasma_pheno$Sample.Number, colnames(plasma_miRNA_data))]</code></pre>
</div>
<div id="normalisation" class="section level3">
<h3>Normalisation</h3>
<p>Manually applied TMM normalisation then converted the counts into
CPM.</p>
<pre class="r"><code>plasma_miRNA_data_DGE &lt;- DGEList(counts = plasma_miRNA_data_raw, lib.size = colSums(plasma_miRNA_data), samples = plasma_pheno)
plasma_miRNA_data_DGE_TMM &lt;- calcNormFactors(plasma_miRNA_data_DGE, method = &quot;TMM&quot;)

raw_counts_CPM &lt;- cpm(plasma_miRNA_data_DGE, log=FALSE)
raw_counts_CPM_TMM &lt;- cpm(plasma_miRNA_data_DGE_TMM, log=FALSE)</code></pre>
</div>
<div id="plot-data" class="section level3">
<h3>Plot data</h3>
<p>Compared the distribution of the data provided to the TMM normalised
CPM values.</p>
<pre class="r"><code>bp_plasma_miRNA_data &lt;- gg_boxplot(plasma_miRNA_data,
                                   log.transform=T,
                                   title=&quot;Plasma miRNA - original CPM data provided&quot;,
                                   xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)



bp_raw_counts_CPM_TMM &lt;- gg_boxplot(raw_counts_CPM_TMM,
                                   log.transform=T,
                                   title=&quot;Plasma miRNA - raw counts, TMM and CPM normalised&quot;,
                                   xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)



bp_plasma_miRNA_data_raw &lt;- gg_boxplot(plasma_miRNA_data_raw,
                                   log.transform=T,
                                   title=&quot;Plasma miRNA - Log2 raw counts&quot;,
                                   xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)

  

bp_raw_counts_CPM &lt;- gg_boxplot(raw_counts_CPM,
                                   log.transform=T,
                                   title=&quot;Plasma miRNA - raw counts, just CPM normalised&quot;,
                                   xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="synovial-fluid" class="section level2">
<h2>Synovial fluid</h2>
<div id="read-data-in-1" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised SF miRNA data
sf_miRNA_data &lt;- read.csv(&quot;./data/miRNA_sf.csv&quot;, row.names = 1)
colnames(sf_miRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(sf_miRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>sf_miRNA_data &lt;- sf_miRNA_data[,match(sf_pheno$Sample.Number, colnames(sf_miRNA_data))]</code></pre>
</div>
<div id="plot-data-1" class="section level3">
<h3>Plot data</h3>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_sf_miRNA_data &lt;- gg_boxplot(sf_miRNA_data,
                                   log.transform=T,
                                   title=&quot;Synovial fluid miRNA - original CPM data provided&quot;,
                                   xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="lncrna" class="section level1">
<h1>lncRNA</h1>
<p>The rest of the sequencing data didn’t have raw counts.The CPM data
provided will be read in for the lncRNA, snoRNA, snRNA, tRNA.</p>
<div id="plasma-1" class="section level2">
<h2>Plasma:</h2>
<div id="read-data-in-2" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised plasma lncRNA data
plasma_lncRNA_data &lt;- read.csv(&quot;./data/lncRNA_plasma.csv&quot;, row.names = 1)
colnames(plasma_lncRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(plasma_lncRNA_data))
row.names(plasma_lncRNA_data) &lt;- gsub(pattern = &quot;_9796&quot;, replacement = &quot;&quot;, x = row.names(plasma_lncRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>plasma_lncRNA_data &lt;- plasma_lncRNA_data[,match(plasma_pheno$Sample.Number, colnames(plasma_lncRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_plasma_lncRNA_data &lt;- gg_boxplot(plasma_lncRNA_data,
                                log.transform =T,
                                title=&quot;Plasma lncRNA&quot;,
                                xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="synovial-fluid-1" class="section level2">
<h2>Synovial fluid:</h2>
<div id="read-data-in-3" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised SF lncRNA data
sf_lncRNA_data &lt;- read.csv(&quot;./data/lncRNA_sf.csv&quot;, row.names = 1)
colnames(sf_lncRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(sf_lncRNA_data))
row.names(sf_lncRNA_data) &lt;- gsub(pattern = &quot;_9796&quot;, replacement = &quot;&quot;, x = row.names(sf_lncRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>sf_lncRNA_data &lt;- sf_lncRNA_data[,match(sf_pheno$Sample.Number, colnames(sf_lncRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_sf_lncRNA_data &lt;- gg_boxplot(sf_lncRNA_data,
                               log.transform=T,
                               title=&quot;Synovial fluid lncRNA&quot;,
                               xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="snorna" class="section level1">
<h1>snoRNA</h1>
<p>The rest of the sequencing data didn’t have raw counts.The CPM data
provided will be read in for the snoRNA, snoRNA, snRNA, tRNA.</p>
<div id="plasma-2" class="section level2">
<h2>Plasma:</h2>
<div id="read-data-in-4" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised plasma snoRNA data
plasma_snoRNA_data &lt;- read.csv(&quot;./data/snoRNA_plasma.csv&quot;, row.names = 1)
colnames(plasma_snoRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(plasma_snoRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>plasma_snoRNA_data &lt;- plasma_snoRNA_data[,match(plasma_pheno$Sample.Number, colnames(plasma_snoRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_plasma_snoRNA_data &lt;- gg_boxplot(plasma_snoRNA_data,
                                log.transform =T,
                                title=&quot;Plasma snoRNA&quot;,
                                xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="synovial-fluid-2" class="section level2">
<h2>Synovial fluid:</h2>
<div id="read-data-in-5" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised SF snoRNA data
sf_snoRNA_data &lt;- read.csv(&quot;./data/snoRNA_sf.csv&quot;, row.names = 1)
colnames(sf_snoRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(sf_snoRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>sf_snoRNA_data &lt;- sf_snoRNA_data[,match(sf_pheno$Sample.Number, colnames(sf_snoRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_sf_snoRNA_data &lt;- gg_boxplot(sf_snoRNA_data,
                               log.transform=T,
                               title=&quot;Synovial fluid snoRNA&quot;,
                               xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="snrna" class="section level1">
<h1>snRNA</h1>
<p>The rest of the sequencing data didn’t have raw counts.The CPM data
provided will be read in for the snRNA, snRNA, snRNA, tRNA.</p>
<div id="plasma-3" class="section level2">
<h2>Plasma:</h2>
<div id="read-data-in-6" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised plasma snRNA data
plasma_snRNA_data &lt;- read.csv(&quot;./data/snRNA_plasma.csv&quot;, row.names = 1)
colnames(plasma_snRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(plasma_snRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>plasma_snRNA_data &lt;- plasma_snRNA_data[,match(plasma_pheno$Sample.Number, colnames(plasma_snRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_plasma_snRNA_data &lt;- gg_boxplot(plasma_snRNA_data,
                                log.transform =T,
                                title=&quot;Plasma snRNA&quot;,
                                xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="synovial-fluid-3" class="section level2">
<h2>Synovial fluid:</h2>
<div id="read-data-in-7" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised SF snRNA data
sf_snRNA_data &lt;- read.csv(&quot;./data/snRNA_sf.csv&quot;, row.names = 1)
colnames(sf_snRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(sf_snRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>sf_snRNA_data &lt;- sf_snRNA_data[,match(sf_pheno$Sample.Number, colnames(sf_snRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_sf_snRNA_data &lt;- gg_boxplot(sf_snRNA_data,
                               log.transform=T,
                               title=&quot;Synovial fluid snRNA&quot;,
                               xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-36-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="trna" class="section level1">
<h1>tRNA</h1>
<p>The rest of the sequencing data didn’t have raw counts.The CPM data
provided will be read in for the tRNA, tRNA, tRNA, tRNA.</p>
<div id="plasma-4" class="section level2">
<h2>Plasma:</h2>
<div id="read-data-in-8" class="section level3">
<h3>Read data in</h3>
<pre class="r"><code># CPM normalised plasma tRNA data
plasma_tRNA_data &lt;- read.csv(&quot;./data/tRNA_plasma.csv&quot;, row.names = 1)
colnames(plasma_tRNA_data) &lt;- gsub(pattern = &quot;X&quot;, replacement = &quot;Sample_&quot;, x = colnames(plasma_tRNA_data))</code></pre>
<p>Process raw data and make sure its in the same order as the phenotype
table:</p>
<pre class="r"><code>plasma_tRNA_data &lt;- plasma_tRNA_data[,match(plasma_pheno$Sample.Number, colnames(plasma_tRNA_data))]</code></pre>
<p>Look at the distribution of the data provided…</p>
<pre class="r"><code>bp_plasma_tRNA_data &lt;- gg_boxplot(plasma_tRNA_data,
                                log.transform =T,
                                title=&quot;Plasma tRNA&quot;,
                                xlab=&quot;Sample ID&quot;, ylab=&quot;Log2 CPM&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="additional-qc---missing-values" class="section level1">
<h1>Additional QC - missing values</h1>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>A problem with this analysis is the high number of zeroes present in
the data. These could be biological zeroes or sampling zeroes. This
problem has been reviewed in depth <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7568192/">elsewhere</a>
<span class="citation">(Silverman et al. 2020)</span>.</p>
<p>It is normal in sequencing analysis to add a small constant to the
count data before log2 transforming it. In the case of sampling zeroes
adding a small constant to all the data before log-transforming would be
inappropriate. The plasma miRNA data will be used to demonstrate this
problem below.</p>
<p>Create two dataframes, one a log transformed matrix with all the
zeroes removed, the other a matrix with a small constant added before
log transformation.</p>
<pre class="r"><code># Log2 transform data then remove the Inf values resulting from log2 transforming 0
# Inf2NA is a custom helper function
plasma_miRNA_data_log2 &lt;- plasma_miRNA_data %&gt;% log2(.) %&gt;% Inf2NA(.)


# Add a small constant to the data then log transform
plasma_miRNA_data_log2_constant &lt;- log2(plasma_miRNA_data+1)</code></pre>
<p>PCAtools requires the row names in the metadata to be the same as the
column names.</p>
<pre class="r"><code>row.names(plasma_pheno) &lt;- plasma_pheno$Sample.Number</code></pre>
<p>Carry out PCA on the log2 transformed data and the log2+1 transformed
data:</p>
<pre class="r"><code># Log2
plasma_miRNA_data_log2_PCA &lt;- pca(na.omit(plasma_miRNA_data_log2), metadata = plasma_pheno, scale = TRUE, center = TRUE)

# Log2+1
plasma_miRNA_data_log2_constant_PCA &lt;- pca(na.omit(plasma_miRNA_data_log2_constant), metadata = plasma_pheno, scale = TRUE, center = TRUE)</code></pre>
<p>Now create PCA plots and correlation plots for the data.</p>
<pre class="r"><code># Log2

# Correlation plot
log2_corplot &lt;- eigencorplotPCA(plasma_miRNA_data_log2_PCA,
    metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;miRNA&quot;)) 


# PCA plot
log2_PCAplot &lt;- plotPCA(plasma_miRNA_data_log2_PCA,
        PCs=c(1,2),
        colour.data=plasma_pheno$Day,
        shape.data=plasma_pheno$Horse,
        colour.lab=&quot;Day&quot;,
        shape.lab=&quot;Horse&quot;)</code></pre>
<pre class="r"><code># Log2 + 1

# Correlation plot
log2_constant_corplot &lt;- eigencorplotPCA(plasma_miRNA_data_log2_constant_PCA,
    metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;miRNA&quot;)) 


# PCA plot
log2_constant_PCAplot &lt;- plotPCA(plasma_miRNA_data_log2_constant_PCA,
        PCs=c(1,2),
        colour.data=plasma_pheno$Day,
        shape.data=plasma_pheno$Horse,
        colour.lab=&quot;Day&quot;,
        shape.lab=&quot;Horse&quot;)</code></pre>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-46-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="pearsons-correlation" class="section level2">
<h2>Pearson’s correlation</h2>
<p>Pearson’s correlation of small RNA sequencing depth vs number of
missing values:</p>
<pre class="r"><code>cor.test(colSums(isZero(plasma_miRNA_data)), plasma_pheno$miRNA, method = &quot;pearson&quot;)</code></pre>
<pre><code>
    Pearson&#39;s product-moment correlation

data:  colSums(isZero(plasma_miRNA_data)) and plasma_pheno$miRNA
t = -17.234, df = 24, p-value = 5.097e-15
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 -0.9829902 -0.9157317
sample estimates:
       cor 
-0.9618909 </code></pre>
<div id="plasma-5" class="section level3">
<h3>Plasma</h3>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-48-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The results demonstrate a relationship between sequencing depth and
missingness. However, that result doesn’t appear to be perfectly
linear.</p>
</div>
<div id="synovial-fluid-4" class="section level3">
<h3>Synovial fluid</h3>
<p><img src="figure/04_Sequencing_QC_normalisation.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="filter-data-by-missingness" class="section level2">
<h2>Filter data by missingness</h2>
<pre class="r"><code># Plasma
plasma_miRNA_data &lt;- plasma_miRNA_data[rowSums(plasma_miRNA_data&gt;0)&gt;=round(ncol(plasma_miRNA_data)*0.6),]
plasma_lncRNA_data &lt;- plasma_lncRNA_data[rowSums(plasma_lncRNA_data&gt;0)&gt;=round(ncol(plasma_lncRNA_data)*0.6),]
plasma_snoRNA_data &lt;- plasma_snoRNA_data[rowSums(plasma_snoRNA_data&gt;0)&gt;=round(ncol(plasma_snoRNA_data)*0.6),]
plasma_snRNA_data &lt;- plasma_snRNA_data[rowSums(plasma_snRNA_data&gt;0)&gt;=round(ncol(plasma_snRNA_data)*0.6),]
plasma_tRNA_data &lt;- plasma_tRNA_data[rowSums(plasma_tRNA_data&gt;0)&gt;=round(ncol(plasma_tRNA_data)*0.6),]

# SF
sf_miRNA_data &lt;- sf_miRNA_data[rowSums(sf_miRNA_data&gt;0)&gt;=round(ncol(sf_miRNA_data)*0.6),]
sf_lncRNA_data &lt;- sf_lncRNA_data[rowSums(sf_lncRNA_data&gt;0)&gt;=round(ncol(sf_lncRNA_data)*0.6),]
sf_snoRNA_data &lt;- sf_snoRNA_data[rowSums(sf_snoRNA_data&gt;0)&gt;=round(ncol(sf_snoRNA_data)*0.6),]
sf_snRNA_data &lt;- sf_snRNA_data[rowSums(sf_snRNA_data&gt;0)&gt;=round(ncol(sf_snRNA_data)*0.6),]</code></pre>
</div>
</div>
<div id="write-r-data-object-outputs" class="section level1">
<h1>Write R data object outputs</h1>
<p>Now that data has been filtered for sampling bias, log2 normalise and
add constant to filtered data:</p>
<pre class="r"><code># Plasma 
saveRDS(plasma_pheno, &quot;./data/plasma_sequencing_pheno.rds&quot;)
saveRDS(log2(plasma_miRNA_data+1), &quot;./data/plasma_miRNA_data.rds&quot;)
saveRDS(log2(plasma_lncRNA_data+1), &quot;./data/plasma_lncRNA_data.rds&quot;)
saveRDS(log2(plasma_snoRNA_data+1), &quot;./data/plasma_snoRNA_data.rds&quot;)
saveRDS(log2(plasma_snRNA_data+1), &quot;./data/plasma_snRNA_data.rds&quot;)
saveRDS(log2(plasma_tRNA_data+1), &quot;./data/plasma_tRNA_data.rds&quot;)

# SF
saveRDS(sf_pheno, &quot;./data/sf_sequencing_pheno.rds&quot;)
saveRDS(log2(sf_miRNA_data+1), &quot;./data/sf_miRNA_data.rds&quot;)
saveRDS(log2(sf_lncRNA_data+1), &quot;./data/sf_lncRNA_data.rds&quot;)
saveRDS(log2(sf_snoRNA_data+1), &quot;./data/sf_snoRNA_data.rds&quot;)
saveRDS(log2(sf_snRNA_data+1), &quot;./data/sf_snRNA_data.rds&quot;)</code></pre>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-silverman2020" class="csl-entry">
Silverman, Justin D., Kimberly Roche, Sayan Mukherjee, and Lawrence A.
David. 2020. <span>“Naught All Zeros in Sequence Count Data Are the
Same.”</span> <em>Computational and Structural Biotechnology
Journal</em> 18 (September): 2789–98. <a
href="https://doi.org/10.1016/j.csbj.2020.09.014">https://doi.org/10.1016/j.csbj.2020.09.014</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.utf8 
[2] LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggpubr_0.4.0      viridis_0.6.2     viridisLite_0.4.0 edgeR_3.38.1     
 [5] limma_3.52.1      R.utils_2.11.0    R.oo_1.25.0       R.methodsS3_1.8.2
 [9] PCAtools_2.8.0    ggrepel_0.9.1     mixOmics_6.20.0   lattice_0.20-45  
[13] MASS_7.3-56       cowplot_1.1.1     reshape2_1.4.4    forcats_0.5.1    
[17] stringr_1.4.0     dplyr_1.0.9       purrr_0.3.4       readr_2.1.2      
[21] tidyr_1.2.0       tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.1  
[25] workflowr_1.7.0  

loaded via a namespace (and not attached):
  [1] colorspace_2.0-3          ggsignif_0.6.3           
  [3] ellipsis_0.3.2            rprojroot_2.0.3          
  [5] corpcor_1.6.10            fs_1.5.2                 
  [7] rstudioapi_0.13           farver_2.1.0             
  [9] RSpectra_0.16-1           fansi_1.0.3              
 [11] lubridate_1.8.0           xml2_1.3.3               
 [13] splines_4.2.0             codetools_0.2-18         
 [15] sparseMatrixStats_1.8.0   knitr_1.39               
 [17] jsonlite_1.8.0            broom_0.8.0              
 [19] dbplyr_2.2.0              compiler_4.2.0           
 [21] httr_1.4.3                dqrng_0.3.0              
 [23] backports_1.4.1           assertthat_0.2.1         
 [25] Matrix_1.5-1              fastmap_1.1.0            
 [27] cli_3.3.0                 later_1.3.0              
 [29] BiocSingular_1.12.0       htmltools_0.5.2          
 [31] tools_4.2.0               rsvd_1.0.5               
 [33] igraph_1.3.2              gtable_0.3.0             
 [35] glue_1.6.2                Rcpp_1.0.8.3             
 [37] carData_3.0-5             cellranger_1.1.0         
 [39] jquerylib_0.1.4           vctrs_0.4.1              
 [41] nlme_3.1-157              DelayedMatrixStats_1.18.0
 [43] xfun_0.31                 ps_1.7.0                 
 [45] beachmat_2.12.0           rvest_1.0.2              
 [47] lifecycle_1.0.1           irlba_2.3.5              
 [49] rstatix_0.7.0             getPass_0.2-2            
 [51] scales_1.2.0              hms_1.1.1                
 [53] promises_1.2.0.1          MatrixGenerics_1.8.0     
 [55] parallel_4.2.0            RColorBrewer_1.1-3       
 [57] yaml_2.3.5                gridExtra_2.3            
 [59] sass_0.4.1                stringi_1.7.6            
 [61] highr_0.9                 S4Vectors_0.34.0         
 [63] ScaledMatrix_1.4.0        BiocGenerics_0.42.0      
 [65] BiocParallel_1.30.3       rlang_1.0.2              
 [67] pkgconfig_2.0.3           matrixStats_0.62.0       
 [69] evaluate_0.15             labeling_0.4.2           
 [71] processx_3.6.0            tidyselect_1.1.2         
 [73] plyr_1.8.7                magrittr_2.0.3           
 [75] R6_2.5.1                  IRanges_2.30.0           
 [77] generics_0.1.2            DelayedArray_0.22.0      
 [79] DBI_1.1.2                 mgcv_1.8-40              
 [81] pillar_1.7.0              haven_2.5.0              
 [83] whisker_0.4               withr_2.5.0              
 [85] abind_1.4-5               car_3.1-0                
 [87] modelr_0.1.8              crayon_1.5.1             
 [89] rARPACK_0.11-0            utf8_1.2.2               
 [91] ellipse_0.4.3             tzdb_0.3.0               
 [93] rmarkdown_2.14            locfit_1.5-9.5           
 [95] grid_4.2.0                readxl_1.4.0             
 [97] callr_3.7.0               git2r_0.30.1             
 [99] reprex_2.0.1              digest_0.6.29            
[101] httpuv_1.6.5              stats4_4.2.0             
[103] munsell_0.5.0             bslib_0.3.1              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
