<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Johnson" />

<meta name="date" content="2022-09-28" />

<title>Exploratory data analysis - sequencing data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Longitudinal OA multi-omics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="02_EDA.html">Exploratory data analysis</a>
    </li>
    <li>
      <a href="03_Differential_expression.html">Differential expression analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Small RNA sequencing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_Sequencing_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="05_Sequencing_EDA.html">Exploratory data analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Multi-omics analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06_jDR.html">Joint dimensionality reduction v1</a>
    </li>
    <li>
      <a href="06_jDR_2.html">Joint dimensionality reduction v2</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="07_Clustering.html">Clustering</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="08_Functional_enrichment.html">Functional enrichment v1</a>
    </li>
    <li>
      <a href="08_Functional_enrichment_2.html">Functional enrichment v2</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ejjohnson93/anderson-peffers_OA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Exploratory data analysis - sequencing
data</h1>
<h4 class="author">Emily Johnson</h4>
<h4 class="date">2022-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>anderson-peffers_OA/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220927code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220927code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220927)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreecbc8d588a2797af629dcbd0a50035fa7ae7026b1targetblankcbc8d58a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/cbc8d588a2797af629dcbd0a50035fa7ae7026b1" target="_blank">cbc8d58</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreecbc8d588a2797af629dcbd0a50035fa7ae7026b1targetblankcbc8d58a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/cbc8d588a2797af629dcbd0a50035fa7ae7026b1" target="_blank">cbc8d58</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/06_jDR_2.Rmd
    Untracked:  analysis/08_Functional_enrichment_2.Rmd
    Untracked:  analysis/about_check.Rmd
    Untracked:  code/modified_heatmap.R
    Untracked:  data/complete_annotation.csv
    Untracked:  data/complete_annotation.rds
    Untracked:  data/miRNA.dat
    Untracked:  data/plasma_sequencing_data.rds
    Untracked:  data/sf_sequencing_data.rds
    Untracked:  imgs/jDR/plasma/v1/
    Untracked:  imgs/jDR/plasma/v2_USETHIS/
    Untracked:  imgs/jDR/synovial_fluid/v1/
    Untracked:  imgs/jDR/synovial_fluid/v2_USETHIS/
    Untracked:  imgs/sampling_types.png
    Untracked:  output/complete_annotation.csv
    Untracked:  output/complete_annotation.rds
    Untracked:  output/enrichment_results/
    Untracked:  plasma_model2.hdf5
    Untracked:  sf_time_model2.hdf5

Unstaged changes:
    Modified:   analysis/04_Sequencing_QC_normalisation.Rmd
    Modified:   analysis/05_Sequencing_EDA.Rmd
    Modified:   analysis/06_jDR.Rmd
    Deleted:    analysis/06_jDR_test.Rmd
    Modified:   analysis/08_Functional_enrichment.Rmd
    Modified:   analysis/_site.yml
    Modified:   data/mofa2_plasma.rds
    Modified:   data/mofa2_plasma_df.rds
    Modified:   data/mofa2_sf.rds
    Modified:   data/mofa2_sf_df.rds
    Deleted:    docs.zip
    Deleted:    imgs/jDR/plasma/gsea_lipid_binding.png
    Deleted:    imgs/jDR/plasma/gsea_organitrogen.png
    Deleted:    imgs/jDR/plasma/gsea_phospholipid.png
    Deleted:    imgs/jDR/plasma/miRNA_plasma_factor2.PDF
    Deleted:    imgs/jDR/plasma/plasma_f2_gsea.PDF
    Deleted:    imgs/jDR/plasma/plasma_f2_gsea.png
    Deleted:    imgs/jDR/plasma/plasma_factor_group_correlation_mofa.PDF
    Deleted:    imgs/jDR/plasma/plasma_factor_variation_mofa.PDF
    Deleted:    imgs/jDR/plasma/plasma_factors_gsea_non-sig.PDF
    Deleted:    imgs/jDR/plasma/plasma_factors_gsea_non-sig.png
    Deleted:    imgs/jDR/plasma/plasma_mofa.PDF
    Deleted:    imgs/jDR/plasma/protein_plasma_factor2.PDF
    Deleted:    imgs/jDR/plasma/synovial_fluid_f5_gsea.PDF
    Deleted:    imgs/jDR/plasma/synovial_fluid_f5_gsea.png
    Deleted:    imgs/jDR/plasma/synovial_fluid_factors_gsea_non-sig.PDF
    Deleted:    imgs/jDR/plasma/synovial_fluid_factors_gsea_non-sig.png
    Deleted:    imgs/jDR/synovial_fluid/f3_gsea_inflammation.png
    Deleted:    imgs/jDR/synovial_fluid/gsea_f5_supramolecular_complex.png
    Deleted:    miRNA.dat
    Modified:   plasma_model.hdf5
    Deleted:    protein_plasma_factor2_names.PDF
    Modified:   sf_model.hdf5
    Modified:   sf_time_model.hdf5

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/05_Sequencing_EDA.Rmd</code>) and
HTML (<code>docs/05_Sequencing_EDA.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/7648b32ce01fe6db04bcff0d91151a4188130207/analysis/05_Sequencing_EDA.Rmd" target="_blank">7648b32</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-12-01
</td>
<td>
Further visualisation updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/7648b32ce01fe6db04bcff0d91151a4188130207/docs/05_Sequencing_EDA.html" target="_blank">7648b32</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-12-01
</td>
<td>
Further visualisation updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/6ed591f12047bc35eb7635f1d01e14badef83ec8/analysis/05_Sequencing_EDA.Rmd" target="_blank">6ed591f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-11
</td>
<td>
jDR + clustering + knitting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/6ed591f12047bc35eb7635f1d01e14badef83ec8/docs/05_Sequencing_EDA.html" target="_blank">6ed591f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-11
</td>
<td>
jDR + clustering + knitting
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/928fd819c7a6726f3ea0bc9bedec6e1910242471/analysis/05_Sequencing_EDA.Rmd" target="_blank">928fd81</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-08
</td>
<td>
jDR + EDA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/928fd819c7a6726f3ea0bc9bedec6e1910242471/docs/05_Sequencing_EDA.html" target="_blank">928fd81</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-08
</td>
<td>
jDR + EDA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/e34c9f9d885837b14673c528fea1cce3b9824a07/analysis/05_Sequencing_EDA.Rmd" target="_blank">e34c9f9</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-03
</td>
<td>
Updated theme + additional analysis docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/c63c64ff60364617182a11660b1352d65c910cac/analysis/05_Sequencing_EDA.Rmd" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/c63c64ff60364617182a11660b1352d65c910cac/docs/05_Sequencing_EDA.html" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>source(&quot;./code/utility-functions.R&quot;) # load helper functions 
library(tidyverse)
library(cowplot) # plot theme
library(mixOmics) # multi-level PCA
library(viridis) # colour schemes
library(ggpubr) # utilities to arrange/annotate ggplot2 plots
library(PCAtools) # pca &amp; correlation plot</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code># Plasma 
plasma_pheno_seq &lt;- readRDS(&quot;./data/plasma_sequencing_pheno.rds&quot;) 
plasma_miRNA_data &lt;- readRDS(&quot;./data/plasma_miRNA_data.rds&quot;) 
plasma_lncRNA_data &lt;- readRDS(&quot;./data/plasma_lncRNA_data.rds&quot;)
plasma_snoRNA_data &lt;- readRDS(&quot;./data/plasma_snoRNA_data.rds&quot;)
plasma_snRNA_data &lt;- readRDS(&quot;./data/plasma_snRNA_data.rds&quot;)
plasma_tRNA_data &lt;- readRDS(&quot;./data/plasma_tRNA_data.rds&quot;)
plasma_sequencing_data &lt;- readRDS(&quot;./data/plasma_sequencing_data.rds&quot;) 

# SF
sf_pheno_seq &lt;- readRDS(&quot;./data/sf_sequencing_pheno.rds&quot;)
sf_miRNA_data &lt;- readRDS(&quot;./data/sf_miRNA_data.rds&quot;)
sf_lncRNA_data &lt;- readRDS(&quot;./data/sf_lncRNA_data.rds&quot;)
sf_snoRNA_data &lt;- readRDS(&quot;./data/sf_snoRNA_data.rds&quot;)
sf_snRNA_data &lt;- readRDS(&quot;./data/sf_snRNA_data.rds&quot;)
sf_sequencing_data &lt;- readRDS(&quot;./data/sf_sequencing_data.rds&quot;)</code></pre>
</div>
<div id="mirna" class="section level1">
<h1>miRNA</h1>
<div id="pca" class="section level2">
<h2>PCA</h2>
<div id="plasma" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code>row.names(plasma_pheno_seq) &lt;- plasma_pheno_seq$Sample.Number</code></pre>
<pre class="r"><code>miRNA_plasma_pca_res &lt;- pca(na.omit(plasma_miRNA_data), metadata = plasma_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># Colour scheme for plasma
redblue &lt;- colorRampPalette(c(&quot;#db092c&quot;, &quot;#984ea3&quot;, &quot;#097ddb&quot;))(50)


# PCA plot
plasma_miRNA_PCA &lt;- plotPCA(miRNA_plasma_pca_res,
        PCs=c(1,2),
        colours = redblue,
        colour.data=plasma_pheno_seq$Day,
        shape.data=plasma_pheno_seq$Horse,
        colour.lab=&quot;Day&quot;,
        shape.lab=&quot;Horse&quot;)

plasma_miRNA_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
plasma_miRNA_corplot &lt;- eigencorplotPCA(miRNA_plasma_pca_res,
    metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;miRNA&quot;)) 


plasma_miRNA_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sf" class="section level3">
<h3>SF</h3>
<pre class="r"><code>row.names(sf_pheno_seq) &lt;- sf_pheno_seq$Sample.Number</code></pre>
<pre class="r"><code>miRNA_sf_pca_res &lt;- pca(na.omit(sf_miRNA_data), metadata = sf_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># Colour scheme for synovial fluid
yellowblue &lt;- hcl.colors(50, &quot;BluYl&quot;)


# PCA plot
sf_miRNA_PCA &lt;- plotPCA(miRNA_sf_pca_res,
                            PCs=c(1,2),
                            colours=yellowblue,
                            colour.data=sf_pheno_seq$Day,
                            shape.data=sf_pheno_seq$Horse,
                            colour.lab=&quot;Day&quot;,
                            shape.lab=&quot;Horse&quot;)

sf_miRNA_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
sf_miRNA_corplot &lt;- eigencorplotPCA(miRNA_sf_pca_res,
                                        metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;miRNA&quot;, &quot;Group&quot;)) 


sf_miRNA_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="multi-level-pca" class="section level2">
<h2>Multi-level PCA</h2>
<div id="plasma-1" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code>plasma_miRNA_ml_PCA &lt;- multi_level_pca(na.omit(plasma_miRNA_data),
                multilevel = plasma_pheno_seq$Horse,
                group = plasma_pheno_seq$Day, group.lab = &quot;Day&quot;, title = &quot;miRNA - Plasma multi-level PCA&quot;)</code></pre>
<pre class="r"><code>plasma_miRNA_ml_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sf-1" class="section level3">
<h3>SF</h3>
<pre class="r"><code>sf_miRNA_ml_PCA &lt;- multi_level_pca(na.omit(sf_miRNA_data),
                multilevel = sf_pheno_seq$Horse,
                group = sf_pheno_seq$Day, group.lab = &quot;Day&quot;, title = &quot;miRNA - Synovial fluid multi-level PCA&quot;)</code></pre>
<pre class="r"><code>sf_miRNA_ml_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="lncrna" class="section level1">
<h1>lncRNA</h1>
<div id="pca-1" class="section level2">
<h2>PCA</h2>
<div id="plasma-2" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code>lncRNA_plasma_pca_res &lt;- pca(na.omit(plasma_lncRNA_data), metadata = plasma_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># PCA plot
plasma_lncRNA_PCA &lt;- plotPCA(lncRNA_plasma_pca_res,
                            PCs=c(1,2),
                            colours=redblue,
                            colour.data=plasma_pheno_seq$Day,
                            shape.data=plasma_pheno_seq$Horse,
                            colour.lab=&quot;Day&quot;,
                            shape.lab=&quot;Horse&quot;)

plasma_lncRNA_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
plasma_lncRNA_corplot &lt;- eigencorplotPCA(lncRNA_plasma_pca_res,
                                        metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;lncRNA&quot;)) 


plasma_lncRNA_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sf-2" class="section level3">
<h3>SF</h3>
<pre class="r"><code>lncRNA_sf_pca_res &lt;- pca(na.omit(sf_lncRNA_data), metadata = sf_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># PCA plot
sf_lncRNA_PCA &lt;- plotPCA(lncRNA_sf_pca_res,
                        PCs=c(1,2),
                        colours=yellowblue,
                        colour.data=sf_pheno_seq$Day,
                        shape.data=sf_pheno_seq$Horse,
                        colour.lab=&quot;Day&quot;,
                        shape.lab=&quot;Horse&quot;)

sf_lncRNA_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
sf_lncRNA_corplot &lt;- eigencorplotPCA(lncRNA_sf_pca_res,
                                    metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;lncRNA&quot;, &quot;Group&quot;)) 


sf_lncRNA_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="multi-level-pca-1" class="section level2">
<h2>Multi-level PCA</h2>
<div id="plasma-3" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code>plasma_lncRNA_ml_PCA &lt;- multi_level_pca(na.omit(plasma_lncRNA_data),
                                       multilevel = plasma_pheno_seq$Horse,
                                       group = plasma_pheno_seq$Day, group.lab = &quot;Day&quot;, title = &quot;lncRNA - Plasma multi-level PCA&quot;)</code></pre>
<pre class="r"><code>plasma_lncRNA_ml_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sf-3" class="section level3">
<h3>SF</h3>
<pre class="r"><code>sf_lncRNA_ml_PCA &lt;- multi_level_pca(na.omit(sf_lncRNA_data),
                                   multilevel = sf_pheno_seq$Horse,
                                   group = sf_pheno_seq$Day, group.lab = &quot;Day&quot;, title = &quot;lncRNA - Synovial fluid multi-level PCA&quot;)</code></pre>
<pre class="r"><code>sf_lncRNA_ml_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="snrna" class="section level1">
<h1>snRNA</h1>
<div id="pca-2" class="section level2">
<h2>PCA</h2>
<div id="plasma-4" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code>snRNA_plasma_pca_res &lt;- pca(na.omit(plasma_snRNA_data), metadata = plasma_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># PCA plot
plasma_snRNA_PCA &lt;- plotPCA(snRNA_plasma_pca_res,
                             PCs=c(1,2),
                             colours=redblue,
                             colour.data=plasma_pheno_seq$Day,
                             shape.data=plasma_pheno_seq$Horse,
                             colour.lab=&quot;Day&quot;,
                             shape.lab=&quot;Horse&quot;)

plasma_snRNA_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
plasma_snRNA_corplot &lt;- eigencorplotPCA(snRNA_plasma_pca_res,
                                         metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;snRNA&quot;)) 


plasma_snRNA_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-22-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sf-4" class="section level3">
<h3>SF</h3>
<pre class="r"><code>snRNA_sf_pca_res &lt;- pca(na.omit(sf_snRNA_data), metadata = sf_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># PCA plot
sf_snRNA_PCA &lt;- plotPCA(snRNA_sf_pca_res,
                         PCs=c(1,2),
                         colours=yellowblue,
                         colour.data=sf_pheno_seq$Day,
                         shape.data=sf_pheno_seq$Horse,
                         colour.lab=&quot;Day&quot;,
                         shape.lab=&quot;Horse&quot;)

sf_snRNA_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-24-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
sf_snRNA_corplot &lt;- eigencorplotPCA(snRNA_sf_pca_res,
                                     metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;snRNA&quot;, &quot;Group&quot;)) 


sf_snRNA_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-24-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="multi-level-pca-2" class="section level2">
<h2>Multi-level PCA</h2>
<div id="plasma-5" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code>plasma_snRNA_ml_PCA &lt;- multi_level_pca(na.omit(plasma_snRNA_data),
                                        multilevel = plasma_pheno_seq$Horse,
                                        group = plasma_pheno_seq$Day, group.lab = &quot;Day&quot;, title = &quot;snRNA - Plasma multi-level PCA&quot;)</code></pre>
<pre class="r"><code>plasma_snRNA_ml_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="sf-5" class="section level3">
<h3>SF</h3>
<pre class="r"><code>sf_snRNA_ml_PCA &lt;- multi_level_pca(na.omit(sf_snRNA_data),
                                    multilevel = sf_pheno_seq$Horse,
                                    group = sf_pheno_seq$Day, group.lab = &quot;Day&quot;, title = &quot;snRNA - Synovial fluid multi-level PCA&quot;)</code></pre>
<pre class="r"><code>sf_snRNA_ml_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="all-sequencing-data" class="section level1">
<h1>All sequencing data</h1>
<pre class="r"><code>seq_plasma_pca_res &lt;- pca(na.omit(plasma_sequencing_data), metadata = plasma_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># PCA plot
plasma_sequencing_PCA &lt;- plotPCA(seq_plasma_pca_res,
                            PCs=c(1,2),
                            colours=redblue,
                            colour.data=plasma_pheno_seq$Day,
                            shape.data=plasma_pheno_seq$Horse,
                            colour.lab=&quot;Day&quot;,
                            shape.lab=&quot;Horse&quot;)

plasma_sequencing_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
plasma_sequencing_corplot &lt;- eigencorplotPCA(seq_plasma_pca_res,
                                        metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;)) 


plasma_sequencing_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-30-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>seq_sf_pca_res &lt;- pca(na.omit(sf_sequencing_data), metadata = sf_pheno_seq, scale = TRUE, center = TRUE)</code></pre>
<pre class="r"><code># PCA plot
sf_sequencing_PCA &lt;- plotPCA(seq_sf_pca_res,
                                 PCs=c(1,3),
                                 colours=yellowblue,
                                 colour.data=sf_pheno_seq$Day,
                                 shape.data=sf_pheno_seq$Horse,
                                 colour.lab=&quot;Day&quot;,
                                 shape.lab=&quot;Horse&quot;)

sf_sequencing_PCA</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Correlation plot
sf_sequencing_corplot &lt;- eigencorplotPCA(seq_sf_pca_res,
                                             metavars = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Date&quot;, &quot;Group&quot;)) 


sf_sequencing_corplot</code></pre>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-32-2.png" width="672" style="display: block; margin: auto;" /></p>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-33-1.png" width="864" style="display: block; margin: auto;" /></p>
<p><img src="figure/05_Sequencing_EDA.Rmd/unnamed-chunk-34-1.png" width="432" style="display: block; margin: auto;" /></p>
</div>
<div id="conclusions" class="section level1">
<h1>Conclusions</h1>
<p>Will not include snoRNA data in clustering analysis due to extremely
high proportion of missing values (only 3 rows in SF data and plasma
data after filtering). However, have included it in factor analysis as
all the sequencing data was collapsed together and factor analysis is
not sensitive to missing values. Notable perhaps that URS00006ACE58_9796
and URS000068B4BE_9796 present in both.</p>
<div id="refs">

</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.utf8 
[2] LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggpubr_0.4.0      viridis_0.6.2     viridisLite_0.4.0 R.utils_2.11.0   
 [5] R.oo_1.25.0       R.methodsS3_1.8.2 PCAtools_2.8.0    ggrepel_0.9.1    
 [9] mixOmics_6.20.0   lattice_0.20-45   MASS_7.3-56       cowplot_1.1.1    
[13] reshape2_1.4.4    forcats_0.5.1     stringr_1.4.0     dplyr_1.0.9      
[17] purrr_0.3.4       readr_2.1.2       tidyr_1.2.0       tibble_3.1.7     
[21] ggplot2_3.3.6     tidyverse_1.3.1  

loaded via a namespace (and not attached):
 [1] colorspace_2.0-3          ggsignif_0.6.3           
 [3] ellipsis_0.3.2            rprojroot_2.0.3          
 [5] corpcor_1.6.10            fs_1.5.2                 
 [7] rstudioapi_0.13           farver_2.1.0             
 [9] RSpectra_0.16-1           fansi_1.0.3              
[11] lubridate_1.8.0           xml2_1.3.3               
[13] codetools_0.2-18          sparseMatrixStats_1.8.0  
[15] knitr_1.39                jsonlite_1.8.0           
[17] workflowr_1.7.0           broom_0.8.0              
[19] dbplyr_2.2.0              compiler_4.2.0           
[21] httr_1.4.3                dqrng_0.3.0              
[23] backports_1.4.1           assertthat_0.2.1         
[25] Matrix_1.5-1              fastmap_1.1.0            
[27] cli_3.3.0                 later_1.3.0              
[29] BiocSingular_1.12.0       htmltools_0.5.2          
[31] tools_4.2.0               rsvd_1.0.5               
[33] igraph_1.3.2              gtable_0.3.0             
[35] glue_1.6.2                Rcpp_1.0.8.3             
[37] carData_3.0-5             cellranger_1.1.0         
[39] jquerylib_0.1.4           vctrs_0.4.1              
[41] DelayedMatrixStats_1.18.0 xfun_0.31                
[43] beachmat_2.12.0           rvest_1.0.2              
[45] lifecycle_1.0.1           irlba_2.3.5              
[47] rstatix_0.7.0             scales_1.2.0             
[49] hms_1.1.1                 promises_1.2.0.1         
[51] MatrixGenerics_1.8.0      parallel_4.2.0           
[53] RColorBrewer_1.1-3        yaml_2.3.5               
[55] gridExtra_2.3             sass_0.4.1               
[57] stringi_1.7.6             highr_0.9                
[59] S4Vectors_0.34.0          ScaledMatrix_1.4.0       
[61] BiocGenerics_0.42.0       BiocParallel_1.30.3      
[63] rlang_1.0.2               pkgconfig_2.0.3          
[65] matrixStats_0.62.0        evaluate_0.15            
[67] labeling_0.4.2            tidyselect_1.1.2         
[69] plyr_1.8.7                magrittr_2.0.3           
[71] R6_2.5.1                  IRanges_2.30.0           
[73] generics_0.1.2            DelayedArray_0.22.0      
[75] DBI_1.1.2                 pillar_1.7.0             
[77] haven_2.5.0               whisker_0.4              
[79] withr_2.5.0               abind_1.4-5              
[81] car_3.1-0                 modelr_0.1.8             
[83] crayon_1.5.1              rARPACK_0.11-0           
[85] utf8_1.2.2                ellipse_0.4.3            
[87] tzdb_0.3.0                rmarkdown_2.14           
[89] grid_4.2.0                readxl_1.4.0             
[91] git2r_0.30.1              reprex_2.0.1             
[93] digest_0.6.29             httpuv_1.6.5             
[95] stats4_4.2.0              munsell_0.5.0            
[97] bslib_0.3.1              </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
