<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Johnson" />

<meta name="date" content="2022-09-28" />

<title>Multi-omics - joint dimensionality reduction</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">anderson-peffers_OA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="02_EDA.html">Exploratory data analysis</a>
    </li>
    <li>
      <a href="03_Differential_expression.html">Differential expression analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Small RNA sequencing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_Sequencing_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="05_Sequencing_EDA.html">Exploratory data analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Multi-omics analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06_jDR.html">Joint dimensionality reduction</a>
    </li>
    <li>
      <a href="07_Clustering.html">Clustering</a>
    </li>
    <li>
      <a href="08_Functional_enrichment.html">Functional enrichment</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ejjohnson93/anderson-peffers_OA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Multi-omics - joint dimensionality
reduction</h1>
<h4 class="author">Emily Johnson</h4>
<h4 class="date">2022-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-11
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>anderson-peffers_OA/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220927code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220927code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220927)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreeb8abb7ebb24e6e1d60f8a8c601795873111a34e1targetblankb8abb7ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/b8abb7ebb24e6e1d60f8a8c601795873111a34e1" target="_blank">b8abb7e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreeb8abb7ebb24e6e1d60f8a8c601795873111a34e1targetblankb8abb7ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/b8abb7ebb24e6e1d60f8a8c601795873111a34e1" target="_blank">b8abb7e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Unstaged changes:
    Modified:   analysis/03_Differential_expression.Rmd
    Modified:   analysis/04_Sequencing_QC_normalisation.Rmd
    Modified:   analysis/06_jDR.Rmd
    Modified:   analysis/07_Clustering.Rmd
    Modified:   analysis/08_Functional_enrichment.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd
    Modified:   code/utility-functions.R
    Modified:   data/plasma_lncRNA_data.rds
    Modified:   data/sf_lncRNA_data.rds
    Modified:   plasma_model.hdf5
    Modified:   sf_model.hdf5

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/06_jDR.Rmd</code>) and HTML
(<code>docs/06_jDR.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/b8abb7ebb24e6e1d60f8a8c601795873111a34e1/analysis/06_jDR.Rmd" target="_blank">b8abb7e</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-09
</td>
<td>
MOFA2
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/b8abb7ebb24e6e1d60f8a8c601795873111a34e1/docs/06_jDR.html" target="_blank">b8abb7e</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-09
</td>
<td>
MOFA2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/928fd819c7a6726f3ea0bc9bedec6e1910242471/analysis/06_jDR.Rmd" target="_blank">928fd81</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-08
</td>
<td>
jDR + EDA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/928fd819c7a6726f3ea0bc9bedec6e1910242471/docs/06_jDR.html" target="_blank">928fd81</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-08
</td>
<td>
jDR + EDA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/e34c9f9d885837b14673c528fea1cce3b9824a07/analysis/06_jDR.Rmd" target="_blank">e34c9f9</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-03
</td>
<td>
Updated theme + additional analysis docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/c63c64ff60364617182a11660b1352d65c910cac/analysis/06_jDR.Rmd" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/c63c64ff60364617182a11660b1352d65c910cac/docs/06_jDR.html" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>source(&quot;./code/utility-functions.R&quot;) # load helper functions 
library(tidyverse)
library(MOFA2) # factor analysis
library(ggpubr) # utilities to arrange/annotate ggplot2 plots
library(cowplot) # plot theme
library(viridis) # colour schemes</code></pre>
</div>
<div id="load-and-process-data" class="section level1">
<h1>Load and process data</h1>
<pre class="r"><code># Plasma 
plasma_annotation &lt;- readRDS(&quot;./data/plasma_proteomics_annotation.rds&quot;)
plasma_pheno &lt;- readRDS(&quot;./data/plasma_pheno.rds&quot;) 
plasma_proteomics_data &lt;- readRDS(&quot;./data/plasma_data_norm.rds&quot;)
plasma_miRNA_data &lt;- readRDS(&quot;./data/plasma_miRNA_data.rds&quot;) 
plasma_lncRNA_data &lt;- readRDS(&quot;./data/plasma_lncRNA_data.rds&quot;)
plasma_snRNA_data &lt;- readRDS(&quot;./data/plasma_snRNA_data.rds&quot;)

# SF
sf_annotation &lt;- readRDS(&quot;./data/sf_proteomics_annotation.rds&quot;)
sf_pheno &lt;- readRDS(&quot;./data/sf_pheno.rds&quot;)
sf_proteomics_data &lt;- readRDS(&quot;./data/sf_data_norm.rds&quot;)
sf_miRNA_data &lt;- readRDS(&quot;./data/sf_miRNA_data.rds&quot;)
sf_lncRNA_data &lt;- readRDS(&quot;./data/sf_lncRNA_data.rds&quot;)
sf_snRNA_data &lt;- readRDS(&quot;./data/sf_snRNA_data.rds&quot;)</code></pre>
<p>Convert 0s to NAs as previous analysis showed they were sampling
zeroes rather than biological zeroes:</p>
<pre class="r"><code># Plasma 
plasma_miRNA_data[plasma_miRNA_data == 0] &lt;- NA
plasma_lncRNA_data[plasma_lncRNA_data == 0] &lt;- NA
plasma_snRNA_data[plasma_snRNA_data == 0] &lt;- NA

# SF
sf_miRNA_data[sf_miRNA_data == 0] &lt;- NA
sf_lncRNA_data[sf_lncRNA_data == 0] &lt;- NA
sf_snRNA_data[sf_snRNA_data == 0] &lt;- NA</code></pre>
<p>Remove universal spike-ins:</p>
<pre class="r"><code>plasma_miRNA_data &lt;- plasma_miRNA_data[-c(1:3),]
sf_miRNA_data &lt;- sf_miRNA_data[-c(1:3),]</code></pre>
</div>
<div id="mofa" class="section level1">
<h1>MOFA</h1>
<div id="plasma" class="section level2">
<h2>Plasma</h2>
<p>Visualuse the distributions of the data:</p>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>MOFA2 needs a column in the phenotype data called ‘sample’. Rename
the ‘Sample_ID’ column to ‘sample’.</p>
<pre class="r"><code>colnames(plasma_pheno)[5] &lt;- &quot;sample&quot;</code></pre>
<p>As an input MOFA2 will take a list containing all the different data.
Create this list:</p>
<pre class="r"><code>plasma_mofa_dat &lt;- list(proteomics = as.matrix(plasma_proteomics_data),
                 miRNA = as.matrix(plasma_miRNA_data),
                 lncRNA = as.matrix(plasma_lncRNA_data),
                 snRNA = as.matrix(plasma_snRNA_data))</code></pre>
<p>Create MOFAobject:</p>
<pre class="r"><code>MOFAobject &lt;- create_mofa(plasma_mofa_dat)</code></pre>
<pre><code>Creating MOFA object from a list of matrices (features as rows, sample as columns)...</code></pre>
<pre class="r"><code>print(MOFAobject)</code></pre>
<pre><code>Untrained MOFA model with the following characteristics: 
 Number of views: 4 
 Views names: proteomics miRNA lncRNA snRNA 
 Number of features (per view): 223 53 29 14 
 Number of groups: 1 
 Groups names: group1 
 Number of samples (per group): 26 
 </code></pre>
<p>The number of features per view can be visualised with
<code>plot_data_overview</code>. Missing samples would be represented as
gaps in the block colours.</p>
<pre class="r"><code>plot_data_overview(MOFAobject)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Set model parameters. Here three objects are created containing the
data options, model options and training options. If the raw counts were
used a poisson distribution would be more appropriate.</p>
<pre class="r"><code># Scale views
data_opts &lt;- get_default_data_options(MOFAobject)
data_opts$scale_views &lt;- TRUE
head(data_opts)</code></pre>
<pre><code>$scale_views
[1] TRUE

$scale_groups
[1] FALSE

$center_groups
[1] TRUE

$use_float32
[1] FALSE

$views
[1] &quot;proteomics&quot; &quot;miRNA&quot;      &quot;lncRNA&quot;     &quot;snRNA&quot;     

$groups
[1] &quot;group1&quot;</code></pre>
<pre class="r"><code># Set number of factors
model_opts &lt;- get_default_model_options(MOFAobject)
model_opts$num_factors &lt;- 6
#model_opts$likelihoods[2:4] &lt;- &quot;poisson&quot;
head(model_opts)</code></pre>
<pre><code>$likelihoods
proteomics      miRNA     lncRNA      snRNA 
&quot;gaussian&quot; &quot;gaussian&quot; &quot;gaussian&quot; &quot;gaussian&quot; 

$num_factors
[1] 6

$spikeslab_factors
[1] FALSE

$spikeslab_weights
[1] TRUE

$ard_factors
[1] FALSE

$ard_weights
[1] TRUE</code></pre>
<pre class="r"><code># Set training options
train_opts &lt;- get_default_training_options(MOFAobject)
train_opts$convergence_mode &lt;- &quot;slow&quot;
train_opts$seed &lt;- 123
train_opts$maxiter &lt;- 3000
head(train_opts)</code></pre>
<pre><code>$maxiter
[1] 3000

$convergence_mode
[1] &quot;slow&quot;

$drop_factor_threshold
[1] -1

$verbose
[1] FALSE

$startELBO
[1] 1

$freqELBO
[1] 5</code></pre>
<p>Prepare MOFA object with modified parameters:</p>
<pre class="r"><code>MOFAobject &lt;- prepare_mofa(
  object = MOFAobject,
  data_options = data_opts,
  model_options = model_opts,
  training_options = train_opts
)</code></pre>
<pre><code>Warning in prepare_mofa(object = MOFAobject, data_options = data_opts,
model_options = model_opts, : Some view(s) have less than 15 features, MOFA will
have little power to to learn meaningful factors for these view(s)....</code></pre>
<pre><code>Checking data options...</code></pre>
<pre><code>Checking training options...</code></pre>
<pre><code>Checking model options...</code></pre>
<p>Train model:</p>
<pre class="r"><code>outfile &lt;- file.path(getwd(),&quot;plasma_model.hdf5&quot;)
MOFAobject.trained &lt;- run_mofa(MOFAobject, outfile, use_basilisk=TRUE)</code></pre>
<p>Add phenotype data to the model:</p>
<pre class="r"><code>samples_metadata(MOFAobject.trained) &lt;- plasma_pheno</code></pre>
<p>Visualise variance explained:</p>
<pre class="r"><code>plot_variance_explained(
  MOFAobject.trained, 
  x = &quot;view&quot;, 
  y = &quot;factor&quot;, 
  plot_total = T
)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Majority is from proteomics data - likely due to the higher number of
features.</p>
<p>Plot correlation between factors:</p>
<pre class="r"><code>plot_factor_cor(MOFAobject.trained)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Correlation between factors and metadata:</p>
<pre class="r"><code>correlate_factors_with_covariates(MOFAobject.trained, 
  covariates = c(&quot;Day&quot;, &quot;Horse&quot;), 
  plot=&quot;log_pval&quot;, 
  alpha=&quot;0.15&quot; # p-value threshold
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>correlate_factors_with_covariates(MOFAobject.trained, 
  covariates = c(&quot;Day&quot;, &quot;Horse&quot;), 
  plot=&quot;r&quot;
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-16-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Factor 2 is significantly correlated with “day”. Factors 5 and 6 are
correlated with horse. Most of the variance in factor 2 is explained by
the proteomics data. The variance in factor 1 mostly comes from the
miRNA and lncRNA data, however that doesn’t appear to correspond to a
biological effect. Factor 3 shows slight correlation with “day”. As the
variance is explained by the proteomics and snRNA data this might also
be interesting to explore.</p>
<p>We can inspect the weights of the different features contributing to
each factor.</p>
<pre class="r"><code>proteomics_f2 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;proteomics&quot;, 
                  factor = 2, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)

# Annotate so Uniprot accessions are replaced by protein names
# The accessions are stored as factors so need to be converted back into character vectors
proteomics_f2$data$feature_id &lt;- plasma_annotation[as.character(proteomics_f2$data$feature_id),]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot weights for factor 2:</p>
<pre class="r"><code>plot_weights(MOFAobject.trained,
 view = &quot;proteomics&quot;,
 factor = 2,
 nfeatures = 50,     # Top number of features to highlight
 scale = F           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_weights(MOFAobject.trained,
 view = &quot;miRNA&quot;,
 factor = 2,
 nfeatures = 25,     # Top number of features to highlight
 scale = F           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-19-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>The distribution of weights for the top ranked proteins contributing
towards factor 2 are heavily negatively skewed. Only Q28377, F7DZE7,
A0A5F5PEG5, and A0A5F5PWX5 are positively associated in the top 50. If
the weights for miRNA are plotted it can be seen their contribution to
factor 2 is low overall, as the weights range from -0.001 to 0.001.</p>
<p>If we plot factor 1 instead, the weight of the miRNAs is much
higher.</p>
<pre class="r"><code>miRNA_f2 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;miRNA&quot;, 
                  factor = 2, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)

miRNA_f1 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;miRNA&quot;, 
                  factor = 1, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>It is interesting that miR-126-5P and miR-126-3P are among the miRNAs
contributing to factor 2. <a
href="https://pubmed.ncbi.nlm.nih.gov/33627637/"
class="uri">https://pubmed.ncbi.nlm.nih.gov/33627637/</a></p>
<p>Visualise the weights for the proteomics and snRNA data contributing
to factor 3:</p>
<pre class="r"><code>snRNA_f3 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;snRNA&quot;, 
                  factor = 3, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)


snRNA_f4 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;snRNA&quot;, 
                  factor = 4, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>proteomics_f3 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;proteomics&quot;, 
                  factor = 3, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)

# Annotate so Uniprot accessions are replaced by protein names
# The accessions are stored as factors so need to be converted back into character vectors
proteomics_f3$data$feature_id &lt;- plasma_annotation[as.character(proteomics_f3$data$feature_id),]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Create heatmap for proteomics data:</p>
<pre class="r"><code># Create annotation for heatmap
my_sample_col &lt;- data.frame(Day = plasma_pheno$Day)
rownames(my_sample_col) &lt;- colnames(plasma_proteomics_data)


plot_data_heatmap(MOFAobject.trained, 
  view = &quot;proteomics&quot;,
  factor = 2,  
  features = 25,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = TRUE,
  scale = &quot;row&quot;,
  denoise = TRUE,
  color = viridis(10),
  border_color = NA
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-26-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot factors 2 and 3 together (equivalent to PCA):</p>
<pre class="r"><code>p &lt;- plot_factors(MOFAobject.trained, 
  factors = c(2,3), 
  color_by = &quot;Day&quot;,
  shape_by = &quot;Horse&quot;,
  dot_size = 4,
  show_missing = T
)

p + theme_cowplot()</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-27-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Project the samples onto factor 2 for each time point.</p>
<pre class="r"><code># Add day as a factor to metadata for visualisation purposes
MOFAobject.trained@samples_metadata$day_group &lt;- as.factor(plasma_pheno$Day)

plot_factor(MOFAobject.trained, 
  factors = 2, 
  color_by = &quot;day_group&quot;,
  add_violin = TRUE,
  dodge = TRUE
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_factor(MOFAobject.trained, 
  factors = 3, 
  color_by = &quot;day_group&quot;,
  add_violin = TRUE,
  dodge = TRUE
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-28-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="synovial-fluid" class="section level2">
<h2>Synovial fluid</h2>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>colnames(sf_pheno)[6] &lt;- &quot;sample&quot;</code></pre>
<pre class="r"><code>sf_mofa_dat &lt;- list(proteomics = as.matrix(sf_proteomics_data),
                        miRNA = as.matrix(sf_miRNA_data),
                        lncRNA = as.matrix(sf_lncRNA_data),
                        snRNA = as.matrix(sf_snRNA_data))</code></pre>
<p>Create MOFAojbect:</p>
<pre class="r"><code>MOFAobject &lt;- create_mofa(sf_mofa_dat)</code></pre>
<pre><code>Creating MOFA object from a list of matrices (features as rows, sample as columns)...</code></pre>
<pre class="r"><code>print(MOFAobject)</code></pre>
<pre><code>Untrained MOFA model with the following characteristics: 
 Number of views: 4 
 Views names: proteomics miRNA lncRNA snRNA 
 Number of features (per view): 622 61 38 15 
 Number of groups: 1 
 Groups names: group1 
 Number of samples (per group): 52 
 </code></pre>
<pre class="r"><code>plot_data_overview(MOFAobject)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-33-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Modify model parameters:</p>
<pre class="r"><code># Scale views
data_opts &lt;- get_default_data_options(MOFAobject)
data_opts$scale_views &lt;- TRUE
head(data_opts)</code></pre>
<pre><code>$scale_views
[1] TRUE

$scale_groups
[1] FALSE

$center_groups
[1] TRUE

$use_float32
[1] FALSE

$views
[1] &quot;proteomics&quot; &quot;miRNA&quot;      &quot;lncRNA&quot;     &quot;snRNA&quot;     

$groups
[1] &quot;group1&quot;</code></pre>
<pre class="r"><code># Set number of factors
model_opts &lt;- get_default_model_options(MOFAobject)
model_opts$num_factors &lt;- 6
head(model_opts)</code></pre>
<pre><code>$likelihoods
proteomics      miRNA     lncRNA      snRNA 
&quot;gaussian&quot; &quot;gaussian&quot; &quot;gaussian&quot; &quot;gaussian&quot; 

$num_factors
[1] 6

$spikeslab_factors
[1] FALSE

$spikeslab_weights
[1] TRUE

$ard_factors
[1] FALSE

$ard_weights
[1] TRUE</code></pre>
<pre class="r"><code># Set training options
train_opts &lt;- get_default_training_options(MOFAobject)
train_opts$convergence_mode &lt;- &quot;slow&quot;
train_opts$seed &lt;- 123
train_opts$maxiter &lt;- 3000
head(train_opts)</code></pre>
<pre><code>$maxiter
[1] 3000

$convergence_mode
[1] &quot;slow&quot;

$drop_factor_threshold
[1] -1

$verbose
[1] FALSE

$startELBO
[1] 1

$freqELBO
[1] 5</code></pre>
<p>Prepare MOFA object with modified parameters:</p>
<pre class="r"><code>MOFAobject &lt;- prepare_mofa(
  object = MOFAobject,
  data_options = data_opts,
  model_options = model_opts,
  training_options = train_opts
)</code></pre>
<pre><code>Checking data options...</code></pre>
<pre><code>Checking training options...</code></pre>
<pre><code>Checking model options...</code></pre>
<p>Train model:</p>
<pre class="r"><code>outfile &lt;- file.path(getwd(),&quot;sf_model.hdf5&quot;)
MOFAobject.trained &lt;- run_mofa(MOFAobject, outfile, use_basilisk=TRUE)</code></pre>
<p>Add phenotype data to the model:</p>
<pre class="r"><code>samples_metadata(MOFAobject.trained) &lt;- sf_pheno

# Add a group-by-day factor 
MOFAobject.trained@samples_metadata$Group_day &lt;- as.factor(paste0(sf_pheno$Group, &quot;_&quot;, sf_pheno$Day))</code></pre>
<p>Visualise variance explained:</p>
<pre class="r"><code>plot_variance_explained(
  MOFAobject.trained, 
  x = &quot;view&quot;, 
  y = &quot;factor&quot;, 
  plot_total = T
)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-38-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-38-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Majority is from proteomics data - likely due to the higher number of
features.</p>
<p>Plot correlation between factors:</p>
<pre class="r"><code>plot_factor_cor(MOFAobject.trained)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-39-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Correlation between factors and metadata:</p>
<pre class="r"><code>correlate_factors_with_covariates(MOFAobject.trained, 
                                  covariates = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Group&quot;, &quot;Group_day&quot;), 
                                  plot=&quot;log_pval&quot;,
                                  alpha=&quot;0.05&quot; # p-value threshold
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-40-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>correlate_factors_with_covariates(MOFAobject.trained, 
  covariates = c(&quot;Day&quot;, &quot;Horse&quot;, &quot;Group&quot;, &quot;Group_day&quot;), 
  plot=&quot;r&quot;
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-40-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Factor 4 is significantly correlated with day, though factors 2-5
seem to have some correlation with day. Factor 4 is mostly explained by
variance in the miRNA. Factor 2 is correlated with the ‘group_day’
variable, though its worth noting this factor had to be coerced into a
numeric variable. For factor 2 miRNA and lncRNA explain more of the
variance.</p>
<p>It may be worth looking at the proteomics weight for factor 1 and 5.
In factor 1’s case the variance is most strongly correlated with the
proteomics but this may be because there is a much higher number of
features. Factor 5 seems to have a slight correlation with the
biological factors of interest.</p>
<pre class="r"><code>miRNA_f2 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;miRNA&quot;, 
                  factor = 2, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)</code></pre>
<pre class="r"><code>lncRNA_f2 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;lncRNA&quot;, 
                  factor = 2, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)</code></pre>
<pre class="r"><code>miRNA_f4 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;miRNA&quot;, 
                  factor = 4, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)</code></pre>
<pre class="r"><code>proteomics_f5 &lt;- plot_top_weights(MOFAobject.trained, 
                  view = &quot;proteomics&quot;, 
                  factor = 3, 
                  nfeatures = 10,
                  scale = F, 
                  abs = T
)


# Annotate so Uniprot accessions are replaced by protein names
# The accessions are stored as factors so need to be converted back into character vectors
proteomics_f5$data$feature_id &lt;- sf_annotation[as.character(proteomics_f5$data$feature_id),]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/06_jDR.Rmd/unnamed-chunk-45-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot weights for factor 2:</p>
<pre class="r"><code>plot_weights(MOFAobject.trained,
 view = &quot;miRNA&quot;,
 factor = 2,
 nfeatures = 20,     # Top number of features to highlight
 scale = F           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-46-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_weights(MOFAobject.trained,
 view = &quot;lncRNA&quot;,
 factor = 2,
 nfeatures = 20,     # Top number of features to highlight
 scale = F           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-46-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_data_heatmap(MOFAobject.trained, 
                  view = &quot;miRNA&quot;,
                  factor = 4,  
                  features = 10,
                  cluster_rows = TRUE, cluster_cols = TRUE,
                  show_rownames = TRUE, show_colnames = TRUE,
                  scale = &quot;row&quot;,
                  denoise = TRUE,
                  color = viridis(10),
                  border_color = NA
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Project the samples onto factor 2 &amp; 4 for each time point.</p>
<pre class="r"><code># Add day as a factor to metadata for visualisation purposes
MOFAobject.trained@samples_metadata$group &lt;- as.factor(paste0(sf_pheno$Group, &quot;_&quot;, sf_pheno$Day))</code></pre>
<pre class="r"><code>plot_factor(MOFAobject.trained, 
  factors = 2, 
  color_by = &quot;group&quot;,
  add_violin = TRUE,
  dodge = TRUE
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-49-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_factor(MOFAobject.trained, 
  factors = 4, 
  color_by = &quot;group&quot;,
  add_violin = TRUE,
  dodge = TRUE
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-50-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="jive" class="section level1">
<h1>JIVE</h1>
</div>
<div id="icluster" class="section level1">
<h1>iCluster</h1>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.utf8 
[2] LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] viridis_0.6.2     viridisLite_0.4.0 ggpubr_0.4.0      MOFA2_1.6.0      
 [5] R.utils_2.11.0    R.oo_1.25.0       R.methodsS3_1.8.2 PCAtools_2.8.0   
 [9] ggrepel_0.9.1     mixOmics_6.20.0   lattice_0.20-45   MASS_7.3-56      
[13] cowplot_1.1.1     reshape2_1.4.4    forcats_0.5.1     stringr_1.4.0    
[17] dplyr_1.0.9       purrr_0.3.4       readr_2.1.2       tidyr_1.2.0      
[21] tibble_3.1.7      ggplot2_3.3.6     tidyverse_1.3.1  

loaded via a namespace (and not attached):
  [1] readxl_1.4.0              backports_1.4.1          
  [3] corrplot_0.92             workflowr_1.7.0          
  [5] plyr_1.8.7                igraph_1.3.2             
  [7] BiocParallel_1.30.3       digest_0.6.29            
  [9] htmltools_0.5.2           fansi_1.0.3              
 [11] magrittr_2.0.3            ScaledMatrix_1.4.0       
 [13] tzdb_0.3.0                modelr_0.1.8             
 [15] matrixStats_0.62.0        rARPACK_0.11-0           
 [17] colorspace_2.0-3          rvest_1.0.2              
 [19] haven_2.5.0               xfun_0.31                
 [21] crayon_1.5.1              jsonlite_1.8.0           
 [23] glue_1.6.2                gtable_0.3.0             
 [25] DelayedArray_0.22.0       car_3.1-0                
 [27] BiocSingular_1.12.0       Rhdf5lib_1.18.2          
 [29] BiocGenerics_0.42.0       HDF5Array_1.24.1         
 [31] abind_1.4-5               scales_1.2.0             
 [33] pheatmap_1.0.12           DBI_1.1.2                
 [35] rstatix_0.7.0             Rcpp_1.0.8.3             
 [37] reticulate_1.25           dqrng_0.3.0              
 [39] rsvd_1.0.5                stats4_4.2.0             
 [41] httr_1.4.3                dir.expiry_1.4.0         
 [43] RColorBrewer_1.1-3        ellipsis_0.3.2           
 [45] pkgconfig_2.0.3           farver_2.1.0             
 [47] sass_0.4.1                uwot_0.1.11              
 [49] dbplyr_2.2.0              utf8_1.2.2               
 [51] here_1.0.1                tidyselect_1.1.2         
 [53] labeling_0.4.2            rlang_1.0.2              
 [55] later_1.3.0               munsell_0.5.0            
 [57] cellranger_1.1.0          tools_4.2.0              
 [59] cli_3.3.0                 generics_0.1.2           
 [61] broom_0.8.0               evaluate_0.15            
 [63] fastmap_1.1.0             yaml_2.3.5               
 [65] knitr_1.39                fs_1.5.2                 
 [67] nlme_3.1-157              sparseMatrixStats_1.8.0  
 [69] whisker_0.4               xml2_1.3.3               
 [71] compiler_4.2.0            rstudioapi_0.13          
 [73] filelock_1.0.2            png_0.1-7                
 [75] ggsignif_0.6.3            reprex_2.0.1             
 [77] bslib_0.3.1               stringi_1.7.6            
 [79] highr_0.9                 basilisk.utils_1.8.0     
 [81] RSpectra_0.16-1           Matrix_1.4-1             
 [83] psych_2.2.5               vctrs_0.4.1              
 [85] pillar_1.7.0              lifecycle_1.0.1          
 [87] rhdf5filters_1.8.0        jquerylib_0.1.4          
 [89] irlba_2.3.5               corpcor_1.6.10           
 [91] httpuv_1.6.5              R6_2.5.1                 
 [93] promises_1.2.0.1          gridExtra_2.3            
 [95] IRanges_2.30.0            codetools_0.2-18         
 [97] assertthat_0.2.1          rhdf5_2.40.0             
 [99] rprojroot_2.0.3           withr_2.5.0              
[101] mnormt_2.1.0              S4Vectors_0.34.0         
[103] parallel_4.2.0            hms_1.1.1                
[105] grid_4.2.0                beachmat_2.12.0          
[107] basilisk_1.8.0            rmarkdown_2.14           
[109] DelayedMatrixStats_1.18.0 MatrixGenerics_1.8.0     
[111] carData_3.0-5             Rtsne_0.16               
[113] git2r_0.30.1              lubridate_1.8.0          
[115] ellipse_0.4.3            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
