<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="ejjohnson93" />

<meta name="date" content="2022-09-28" />

<title>Multi-omics - joint dimensionality reduction</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">anderson-peffers_OA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="01_QC_normalisation.html">QC and normalisation</a>
</li>
<li>
  <a href="02_EDA.html">EDA</a>
</li>
<li>
  <a href="03_DE_and_clustering.html">DE</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ejjohnson93/anderson-peffers_OA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Multi-omics - joint dimensionality
reduction</h1>
<h4 class="author">ejjohnson93</h4>
<h4 class="date">2022-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-08
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>anderson-peffers_OA/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220927code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220927code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220927)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreeae6f5f4c77d2458fec583f75b216e34d5a9478adtargetblankae6f5f4a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/ae6f5f4c77d2458fec583f75b216e34d5a9478ad" target="_blank">ae6f5f4</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreeae6f5f4c77d2458fec583f75b216e34d5a9478adtargetblankae6f5f4a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/ae6f5f4c77d2458fec583f75b216e34d5a9478ad" target="_blank">ae6f5f4</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/images/
    Untracked:  analysis/references.bib
    Untracked:  external_imgs/
    Untracked:  model.hdf5
    Untracked:  plasma_model.hdf5

Unstaged changes:
    Modified:   analysis/04_Sequencing_QC_normalisation.Rmd
    Modified:   analysis/05_Sequencing_EDA.Rmd
    Modified:   analysis/06_jDR.Rmd
    Modified:   code/utility-functions.R
    Modified:   data/plasma_lncRNA_data.rds
    Modified:   data/plasma_miRNA_data.rds
    Modified:   data/plasma_sequencing_info.csv
    Modified:   data/plasma_sequencing_pheno.rds
    Modified:   data/plasma_snRNA_data.rds
    Modified:   data/plasma_snoRNA_data.rds
    Modified:   data/plasma_tRNA_data.rds
    Modified:   data/sf_lncRNA_data.rds
    Modified:   data/sf_miRNA_data.rds
    Modified:   data/sf_sequencing_pheno.rds
    Modified:   data/sf_snRNA_data.rds
    Modified:   data/sf_snoRNA_data.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/06_jDR.Rmd</code>) and HTML
(<code>docs/06_jDR.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/e34c9f9d885837b14673c528fea1cce3b9824a07/analysis/06_jDR.Rmd" target="_blank">e34c9f9</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-03
</td>
<td>
Updated theme + additional analysis docs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/c63c64ff60364617182a11660b1352d65c910cac/analysis/06_jDR.Rmd" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/c63c64ff60364617182a11660b1352d65c910cac/docs/06_jDR.html" target="_blank">c63c64f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-01
</td>
<td>
Sequencing files created
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(MOFA2)</code></pre>
<pre><code>
Attaching package: &#39;MOFA2&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    predict</code></pre>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<pre class="r"><code># Plasma 
plasma_pheno &lt;- readRDS(&quot;./data/plasma_pheno.rds&quot;) 
plasma_proteomics_data &lt;- readRDS(&quot;./data/plasma_data_norm.rds&quot;)
plasma_miRNA_data &lt;- readRDS(&quot;./data/plasma_miRNA_data.rds&quot;) 
plasma_lncRNA_data &lt;- readRDS(&quot;./data/plasma_lncRNA_data.rds&quot;)
plasma_snRNA_data &lt;- readRDS(&quot;./data/plasma_snRNA_data.rds&quot;)

# SF
sf_pheno &lt;- readRDS(&quot;./data/sf_pheno.rds&quot;)
sf_proteomics_data &lt;- readRDS(&quot;./data/sf_data_norm.rds&quot;)
sf_miRNA_data &lt;- readRDS(&quot;./data/sf_miRNA_data.rds&quot;)
sf_lncRNA_data &lt;- readRDS(&quot;./data/sf_lncRNA_data.rds&quot;)
sf_snRNA_data &lt;- readRDS(&quot;./data/sf_snRNA_data.rds&quot;)</code></pre>
</div>
<div id="mofa" class="section level1">
<h1>MOFA</h1>
<div id="plasma" class="section level2">
<h2>Plasma</h2>
<pre class="r"><code>colnames(plasma_pheno)[5] &lt;- &quot;sample&quot;</code></pre>
<pre class="r"><code>plasma_mofa_dat &lt;- list(proteomics = as.matrix(plasma_proteomics_data),
                 miRNA = as.matrix(plasma_miRNA_data),
                 lncRNA = as.matrix(plasma_lncRNA_data),
                 snRNA = as.matrix(plasma_snRNA_data))</code></pre>
<p>Create MOFAojbect:</p>
<pre class="r"><code>MOFAobject &lt;- create_mofa(plasma_mofa_dat)</code></pre>
<pre><code>Creating MOFA object from a list of matrices (features as rows, sample as columns)...</code></pre>
<pre class="r"><code>print(MOFAobject)</code></pre>
<pre><code>Untrained MOFA model with the following characteristics: 
 Number of views: 4 
 Views names: proteomics miRNA lncRNA snRNA 
 Number of features (per view): 223 56 29 14 
 Number of groups: 1 
 Groups names: group1 
 Number of samples (per group): 26 
 </code></pre>
<pre class="r"><code>plot_data_overview(MOFAobject)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Modify model parameters:</p>
<pre class="r"><code># Scale views
data_opts &lt;- get_default_data_options(MOFAobject)
data_opts$scale_views &lt;- TRUE
head(data_opts)</code></pre>
<pre><code>$scale_views
[1] TRUE

$scale_groups
[1] FALSE

$center_groups
[1] TRUE

$use_float32
[1] FALSE

$views
[1] &quot;proteomics&quot; &quot;miRNA&quot;      &quot;lncRNA&quot;     &quot;snRNA&quot;     

$groups
[1] &quot;group1&quot;</code></pre>
<pre class="r"><code># Set number of factors
model_opts &lt;- get_default_model_options(MOFAobject)
model_opts$num_factors &lt;- 6
head(model_opts)</code></pre>
<pre><code>$likelihoods
proteomics      miRNA     lncRNA      snRNA 
&quot;gaussian&quot; &quot;gaussian&quot; &quot;gaussian&quot; &quot;gaussian&quot; 

$num_factors
[1] 6

$spikeslab_factors
[1] FALSE

$spikeslab_weights
[1] TRUE

$ard_factors
[1] FALSE

$ard_weights
[1] TRUE</code></pre>
<pre class="r"><code># Set training options
train_opts &lt;- get_default_training_options(MOFAobject)
train_opts$convergence_mode &lt;- &quot;slow&quot;
head(train_opts)</code></pre>
<pre><code>$maxiter
[1] 1000

$convergence_mode
[1] &quot;slow&quot;

$drop_factor_threshold
[1] -1

$verbose
[1] FALSE

$startELBO
[1] 1

$freqELBO
[1] 5</code></pre>
<p>Prepare MOFA object with modified parameters:</p>
<pre class="r"><code>MOFAobject &lt;- prepare_mofa(
  object = MOFAobject,
  data_options = data_opts,
  model_options = model_opts,
  training_options = train_opts
)</code></pre>
<pre><code>Warning in prepare_mofa(object = MOFAobject, data_options = data_opts,
model_options = model_opts, : Some view(s) have less than 15 features, MOFA will
have little power to to learn meaningful factors for these view(s)....</code></pre>
<pre><code>Checking data options...</code></pre>
<pre><code>Checking training options...</code></pre>
<pre><code>Checking model options...</code></pre>
<p>Train model:</p>
<pre class="r"><code>outfile &lt;- file.path(getwd(),&quot;plasma_model.hdf5&quot;)
MOFAobject.trained &lt;- run_mofa(MOFAobject, outfile)</code></pre>
<pre><code>Warning: Output file C:/Users/Emily/Documents/GitHub/anderson-peffers_OA/plasma_model.hdf5 already exists, it will be replaced</code></pre>
<pre><code>Connecting to the mofapy2 python package using reticulate (use_basilisk = FALSE)... 
    Please make sure to manually specify the right python binary when loading R with reticulate::use_python(..., force=TRUE) or the right conda environment with reticulate::use_condaenv(..., force=TRUE)
    If you prefer to let us automatically install a conda environment with &#39;mofapy2&#39; installed using the &#39;basilisk&#39; package, please use the argument &#39;use_basilisk = TRUE&#39;</code></pre>
<pre><code>+ &quot;C:/Users/Emily/AppData/Local/r-miniconda/condabin/conda.bat&quot; &quot;create&quot; &quot;--yes&quot; &quot;--prefix&quot; &quot;C:/Users/Emily/AppData/Local/r-miniconda/envs/r-reticulate&quot; &quot;python=3.8&quot; &quot;numpy&quot; &quot;--quiet&quot; &quot;-c&quot; &quot;conda-forge&quot;</code></pre>
<pre><code>Warning in run_mofa(MOFAobject, outfile): mofapy2_0.6.4 is not detected in the
specified python binary, see reticulate::py_config(). Setting use_basilisk =
TRUE...</code></pre>
<pre><code>Connecting to the mofapy2 package using basilisk. 
    Set &#39;use_basilisk&#39; to FALSE if you prefer to manually set the python binary using &#39;reticulate&#39;.</code></pre>
<pre><code>Warning in .quality_control(object, verbose = verbose): Factor(s) 1, 2 are strongly correlated with the total number of expressed features for at least one of your omics. Such factors appear when there are differences in the total &#39;levels&#39; between your samples, *sometimes* because of poor normalisation in the preprocessing steps.</code></pre>
<p>Add phenotype data to the model:</p>
<pre class="r"><code>samples_metadata(MOFAobject.trained) &lt;- plasma_pheno</code></pre>
<p>Visualise variance explained:</p>
<pre class="r"><code>plot_variance_explained(
  MOFAobject.trained, 
  x = &quot;view&quot;, 
  y = &quot;factor&quot;, 
  plot_total = T
)</code></pre>
<pre><code>[[1]]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code>
[[2]]</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-11-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Majority is from proteomics data - likely due to the higher number of
features.</p>
<p>Plot correlation between factors:</p>
<pre class="r"><code>plot_factor_cor(MOFAobject.trained)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Correlation between factors and metadata:</p>
<pre class="r"><code>correlate_factors_with_covariates(MOFAobject.trained, 
  covariates = c(&quot;Day&quot;, &quot;Horse&quot;), 
  plot=&quot;log_pval&quot;
)</code></pre>
<pre><code>Warning in correlate_factors_with_covariates(MOFAobject.trained, covariates =
c(&quot;Day&quot;, : There are non-numeric values in the covariates data.frame, converting
to numeric...</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Factor 1 is correlated with day. Factors 5 and 6 are correlated with
horse.</p>
<pre class="r"><code>plot_top_weights(MOFAobject.trained, 
                  view = &quot;proteomics&quot;, 
                  factor = 1, 
                  nfeatures = 5,
                  scale = T, 
                  abs = T
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_weights(MOFAobject.trained,
 #view = &quot;proteomics&quot;,
 factor = 1,
 nfeatures = 15,     # Top number of features to highlight
 scale = T           # Scale weights from -1 to 1
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_top_weights(MOFAobject.trained, 
                  view = &quot;miRNA&quot;, 
                  factor = 2, 
                  nfeatures = 5,
                  scale = T, 
                  abs = T
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_factors(MOFAobject.trained, 
  factors = c(1,2,3),
  color_by = &quot;Day&quot;
) </code></pre>
<pre><code>Registered S3 method overwritten by &#39;GGally&#39;:
  method from   
  +.gg   ggplot2</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot_data_heatmap(MOFAobject.trained, 
  view = &quot;proteomics&quot;,
  factor = 1,  
  features = 25,
  cluster_rows = TRUE, cluster_cols = TRUE,
  show_rownames = TRUE, show_colnames = TRUE,
  scale = &quot;row&quot;,
  denoise = TRUE
)</code></pre>
<p><img src="figure/06_jDR.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p &lt;- plot_factors(MOFAobject.trained, 
  factors = c(1,2), 
  color_by = &quot;Day&quot;,
  shape_by = &quot;Horse&quot;,
  dot_size = 2.5,
  show_missing = T
)

p &lt;- p + 
  geom_hline(yintercept=-1, linetype=&quot;dashed&quot;) +
  geom_vline(xintercept=(-0.5), linetype=&quot;dashed&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United Kingdom.utf8 
[2] LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] MOFA2_1.6.0     forcats_0.5.1   stringr_1.4.0   dplyr_1.0.9    
 [5] purrr_0.3.4     readr_2.1.2     tidyr_1.2.0     tibble_3.1.7   
 [9] ggplot2_3.3.6   tidyverse_1.3.1

loaded via a namespace (and not attached):
  [1] nlme_3.1-157         matrixStats_0.62.0   fs_1.5.2            
  [4] lubridate_1.8.0      filelock_1.0.2       RColorBrewer_1.1-3  
  [7] httr_1.4.3           rprojroot_2.0.3      tools_4.2.0         
 [10] backports_1.4.1      bslib_0.3.1          utf8_1.2.2          
 [13] R6_2.5.1             HDF5Array_1.24.1     uwot_0.1.11         
 [16] DBI_1.1.2            BiocGenerics_0.42.0  colorspace_2.0-3    
 [19] rhdf5filters_1.8.0   withr_2.5.0          GGally_2.1.2        
 [22] mnormt_2.1.0         tidyselect_1.1.2     compiler_4.2.0      
 [25] git2r_0.30.1         cli_3.3.0            rvest_1.0.2         
 [28] basilisk.utils_1.8.0 xml2_1.3.3           DelayedArray_0.22.0 
 [31] labeling_0.4.2       sass_0.4.1           scales_1.2.0        
 [34] psych_2.2.5          rappdirs_0.3.3       digest_0.6.29       
 [37] rmarkdown_2.14       basilisk_1.8.0       pkgconfig_2.0.3     
 [40] htmltools_0.5.2      MatrixGenerics_1.8.0 highr_0.9           
 [43] dbplyr_2.2.0         fastmap_1.1.0        rlang_1.0.2         
 [46] readxl_1.4.0         rstudioapi_0.13      farver_2.1.0        
 [49] jquerylib_0.1.4      generics_0.1.2       jsonlite_1.8.0      
 [52] magrittr_2.0.3       Matrix_1.4-1         Rhdf5lib_1.18.2     
 [55] Rcpp_1.0.8.3         munsell_0.5.0        S4Vectors_0.34.0    
 [58] fansi_1.0.3          reticulate_1.25      lifecycle_1.0.1     
 [61] stringi_1.7.6        whisker_0.4          yaml_2.3.5          
 [64] Rtsne_0.16           rhdf5_2.40.0         plyr_1.8.7          
 [67] grid_4.2.0           ggrepel_0.9.1        parallel_4.2.0      
 [70] promises_1.2.0.1     crayon_1.5.1         dir.expiry_1.4.0    
 [73] lattice_0.20-45      haven_2.5.0          cowplot_1.1.1       
 [76] hms_1.1.1            knitr_1.39           pillar_1.7.0        
 [79] reshape2_1.4.4       stats4_4.2.0         reprex_2.0.1        
 [82] glue_1.6.2           evaluate_0.15        modelr_0.1.8        
 [85] png_0.1-7            vctrs_0.4.1          tzdb_0.3.0          
 [88] httpuv_1.6.5         cellranger_1.1.0     gtable_0.3.0        
 [91] reshape_0.8.9        assertthat_0.2.1     xfun_0.31           
 [94] broom_0.8.0          later_1.3.0          pheatmap_1.0.12     
 [97] IRanges_2.30.0       corrplot_0.92        workflowr_1.7.0     
[100] ellipsis_0.3.2       here_1.0.1          </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
