<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emily Johnson" />

<meta name="date" content="2022-09-28" />

<title>Multi-omics - clustering</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Longitudinal OA multi-omics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="02_EDA.html">Exploratory data analysis</a>
    </li>
    <li>
      <a href="03_Differential_expression.html">Differential expression analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Small RNA sequencing
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="04_Sequencing_QC_normalisation.html">QC and normalisation</a>
    </li>
    <li>
      <a href="05_Sequencing_EDA.html">Exploratory data analysis</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Multi-omics analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06_jDR.html">Joint dimensionality reduction v1</a>
    </li>
    <li>
      <a href="06_jDR_2.html">Joint dimensionality reduction v2</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="07_Clustering.html">Clustering</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="08_Functional_enrichment.html">Functional enrichment v1</a>
    </li>
    <li>
      <a href="08_Functional_enrichment_2.html">Functional enrichment v2</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ejjohnson93/anderson-peffers_OA">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Multi-omics - clustering</h1>
<h4 class="author">Emily Johnson</h4>
<h4 class="date">2022-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-12-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>anderson-peffers_OA/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220927code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220927code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220927)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreecbc8d588a2797af629dcbd0a50035fa7ae7026b1targetblankcbc8d58a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/cbc8d588a2797af629dcbd0a50035fa7ae7026b1" target="_blank">cbc8d58</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomCBFLivUniandersonpeffersOAtreecbc8d588a2797af629dcbd0a50035fa7ae7026b1targetblankcbc8d58a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/tree/cbc8d588a2797af629dcbd0a50035fa7ae7026b1" target="_blank">cbc8d58</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.Rhistory

Untracked files:
    Untracked:  analysis/06_jDR_2.Rmd
    Untracked:  analysis/08_Functional_enrichment_2.Rmd
    Untracked:  analysis/about_check.Rmd
    Untracked:  code/modified_heatmap.R
    Untracked:  data/complete_annotation.csv
    Untracked:  data/complete_annotation.rds
    Untracked:  data/miRNA.dat
    Untracked:  data/mofa2_plasma_2.rds
    Untracked:  data/mofa2_plasma_df_2.rds
    Untracked:  data/mofa2_sf_2.rds
    Untracked:  data/mofa2_sf_df_2.rds
    Untracked:  data/plasma_sequencing_data.rds
    Untracked:  data/sf_sequencing_data.rds
    Untracked:  imgs/jDR/plasma/v1/
    Untracked:  imgs/jDR/plasma/v2_USETHIS/
    Untracked:  imgs/jDR/synovial_fluid/v1/
    Untracked:  imgs/jDR/synovial_fluid/v2_USETHIS/
    Untracked:  imgs/sampling_types.png
    Untracked:  output/complete_annotation.csv
    Untracked:  output/complete_annotation.rds
    Untracked:  output/enrichment_results/
    Untracked:  plasma_model2.hdf5
    Untracked:  sf_time_model2.hdf5

Unstaged changes:
    Modified:   analysis/04_Sequencing_QC_normalisation.Rmd
    Modified:   analysis/05_Sequencing_EDA.Rmd
    Modified:   analysis/06_jDR.Rmd
    Deleted:    analysis/06_jDR_test.Rmd
    Modified:   analysis/08_Functional_enrichment.Rmd
    Modified:   analysis/_site.yml
    Deleted:    docs.zip
    Deleted:    imgs/jDR/plasma/gsea_lipid_binding.png
    Deleted:    imgs/jDR/plasma/gsea_organitrogen.png
    Deleted:    imgs/jDR/plasma/gsea_phospholipid.png
    Deleted:    imgs/jDR/plasma/miRNA_plasma_factor2.PDF
    Deleted:    imgs/jDR/plasma/plasma_f2_gsea.PDF
    Deleted:    imgs/jDR/plasma/plasma_f2_gsea.png
    Deleted:    imgs/jDR/plasma/plasma_factor_group_correlation_mofa.PDF
    Deleted:    imgs/jDR/plasma/plasma_factor_variation_mofa.PDF
    Deleted:    imgs/jDR/plasma/plasma_factors_gsea_non-sig.PDF
    Deleted:    imgs/jDR/plasma/plasma_factors_gsea_non-sig.png
    Deleted:    imgs/jDR/plasma/plasma_mofa.PDF
    Deleted:    imgs/jDR/plasma/protein_plasma_factor2.PDF
    Deleted:    imgs/jDR/plasma/synovial_fluid_f5_gsea.PDF
    Deleted:    imgs/jDR/plasma/synovial_fluid_f5_gsea.png
    Deleted:    imgs/jDR/plasma/synovial_fluid_factors_gsea_non-sig.PDF
    Deleted:    imgs/jDR/plasma/synovial_fluid_factors_gsea_non-sig.png
    Deleted:    imgs/jDR/synovial_fluid/f3_gsea_inflammation.png
    Deleted:    imgs/jDR/synovial_fluid/gsea_f5_supramolecular_complex.png
    Deleted:    miRNA.dat
    Modified:   output/plasma_limma_DE_results_log2.xlsx
    Modified:   output/plasma_proDA_DE_results.xlsx
    Modified:   output/sf_proDA_DE_results.xlsx
    Modified:   plasma_model.hdf5
    Deleted:    protein_plasma_factor2_names.PDF
    Modified:   sf_model.hdf5
    Modified:   sf_time_model.hdf5

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/07_Clustering.Rmd</code>) and
HTML (<code>docs/07_Clustering.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/526bc37f7b15e35117822b254db15be96996cbd7/analysis/07_Clustering.Rmd" target="_blank">526bc37</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-25
</td>
<td>
Project completion
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/526bc37f7b15e35117822b254db15be96996cbd7/docs/07_Clustering.html" target="_blank">526bc37</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-25
</td>
<td>
Project completion
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/fc9ca8a4449562942f7261173f6140fb19879267/analysis/07_Clustering.Rmd" target="_blank">fc9ca8a</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-23
</td>
<td>
functional enrichment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/218071a73bd878d999f9460324e0b8314c5cc717/analysis/07_Clustering.Rmd" target="_blank">218071a</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-17
</td>
<td>
Clustering + figures
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/218071a73bd878d999f9460324e0b8314c5cc717/docs/07_Clustering.html" target="_blank">218071a</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-17
</td>
<td>
Clustering + figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/2238906ecb478a5d2a9b5529c69c506982c91920/analysis/07_Clustering.Rmd" target="_blank">2238906</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-16
</td>
<td>
Optimising cluster visualisation
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/854584495bb1ae924c0d7f7c02f3b2941f82db32/analysis/07_Clustering.Rmd" target="_blank">8545844</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-15
</td>
<td>
Clustering
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/6ed591f12047bc35eb7635f1d01e14badef83ec8/analysis/07_Clustering.Rmd" target="_blank">6ed591f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-11
</td>
<td>
jDR + clustering + knitting
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/CBFLivUni/anderson-peffers_OA/6ed591f12047bc35eb7635f1d01e14badef83ec8/docs/07_Clustering.html" target="_blank">6ed591f</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-11
</td>
<td>
jDR + clustering + knitting
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/CBFLivUni/anderson-peffers_OA/blob/e34c9f9d885837b14673c528fea1cce3b9824a07/analysis/07_Clustering.Rmd" target="_blank">e34c9f9</a>
</td>
<td>
ejjohnson93
</td>
<td>
2022-11-03
</td>
<td>
Updated theme + additional analysis docs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-libraries" class="section level1">
<h1>Load libraries</h1>
<pre class="r"><code>library(tidyverse)
library(Mfuzz) # clustering package
library(dtwclust) # clustering package
library(cowplot) # ggplot2 theme
library(ggpubr) # utilities to arrange/annotate ggplot2 plots
library(MetBrewer) # additional palettes
library(WGCNA) # matchLabels and accuracyMeasures functions
library(pheatmap) # for correlation plots</code></pre>
</div>
<div id="load-and-process-data" class="section level1">
<h1>Load and process data</h1>
<pre class="r"><code># Plasma 
plasma_annotation &lt;- readRDS(&quot;./data/plasma_proteomics_annotation.rds&quot;)
plasma_pheno &lt;- readRDS(&quot;./data/plasma_pheno.rds&quot;) 
plasma_proteomics_data &lt;- readRDS(&quot;./data/plasma_data_norm.rds&quot;)
plasma_miRNA_data &lt;- readRDS(&quot;./data/plasma_miRNA_data.rds&quot;) 
plasma_lncRNA_data &lt;- readRDS(&quot;./data/plasma_lncRNA_data.rds&quot;)
plasma_snRNA_data &lt;- readRDS(&quot;./data/plasma_snRNA_data.rds&quot;)

# SF
sf_annotation &lt;- readRDS(&quot;./data/sf_proteomics_annotation.rds&quot;)
sf_pheno &lt;- readRDS(&quot;./data/sf_pheno.rds&quot;)
sf_proteomics_data &lt;- readRDS(&quot;./data/sf_data_norm.rds&quot;)
sf_miRNA_data &lt;- readRDS(&quot;./data/sf_miRNA_data.rds&quot;)
sf_lncRNA_data &lt;- readRDS(&quot;./data/sf_lncRNA_data.rds&quot;)
sf_snRNA_data &lt;- readRDS(&quot;./data/sf_snRNA_data.rds&quot;)</code></pre>
<p>Before clustering can be carried out several additional data cleaning
and normalisation steps needed to be carried out. The 0s were first
converted to NAs, as the previous analysis showed they were sampling
zeroes rather than biological zeroes:</p>
<pre class="r"><code># Plasma 
plasma_miRNA_data[plasma_miRNA_data == 0] &lt;- NA
plasma_lncRNA_data[plasma_lncRNA_data == 0] &lt;- NA
plasma_snRNA_data[plasma_snRNA_data == 0] &lt;- NA

# SF
sf_miRNA_data[sf_miRNA_data == 0] &lt;- NA
sf_lncRNA_data[sf_lncRNA_data == 0] &lt;- NA
sf_snRNA_data[sf_snRNA_data == 0] &lt;- NA</code></pre>
<p>Additionally, for the synovial fluid the control samples need to be
removed as these will interfere will the time series clustering.</p>
<pre class="r"><code>sf_pheno &lt;- sf_pheno[!sf_pheno$Group == &quot;Control&quot;,]

sf_proteomics_data &lt;- sf_proteomics_data[,sf_pheno$Sample.Number]
sf_miRNA_data &lt;- sf_miRNA_data[,sf_pheno$Sample.Number]
sf_lncRNA_data &lt;- sf_lncRNA_data[,sf_pheno$Sample.Number]
sf_snRNA_data &lt;- sf_snRNA_data[,sf_pheno$Sample.Number]</code></pre>
<div id="variance-filtering" class="section level2">
<h2>Variance filtering</h2>
<p>Before Z-score normalisation the data was filtered to only include
features with high variance. This is because protein/sequencing data is
extremely noisy. Noisy features can then be assigned to clusters even if
their expression pattern isn’t representative of the cluster.</p>
<p>Plasma:</p>
<pre class="r"><code>## Create vectors of variance/mean value per protein
vars &lt;- apply(plasma_proteomics_data, 1, var, na.rm=TRUE)
means &lt;- apply(plasma_proteomics_data, 1, mean, na.rm=TRUE)

## Subset the data to only include proteins with a variance &gt;1 and mean &gt;2
plasma_proteomics_data &lt;- plasma_proteomics_data[which(vars &gt; 1 &amp; means &gt;2),]

## Remove any rows that contain &gt;20% missing values
plasma_proteomics_data &lt;- plasma_proteomics_data[which(rowMeans(is.na(plasma_proteomics_data)) &lt; 0.2), ]</code></pre>
<p>Synovial fluid:</p>
<pre class="r"><code>## Create vectors of variance/mean value per protein
vars &lt;- apply(sf_proteomics_data, 1, var, na.rm=TRUE)
means &lt;- apply(sf_proteomics_data, 1, mean, na.rm=TRUE)

## Subset the data to only include proteins with a variance &gt;1 and mean &gt;2
sf_proteomics_data &lt;- sf_proteomics_data[which(vars &gt; 1 &amp; means &gt;2),]

## Remove any rows that contain &gt;20% missing values
sf_proteomics_data &lt;- sf_proteomics_data[which(rowMeans(is.na(sf_proteomics_data)) &lt; 0.2), ]</code></pre>
</div>
<div id="z-score-normalisation" class="section level2">
<h2>Z-score normalisation</h2>
<p>Z-score normalisation scales the data such that the mean of all of
the values is 0 and the standard deviation is 1. It is useful for
integrating multi-omics data. This was carried out for the proteomics
and sequencing data, before concatenating the normalised -omics together
for the plasma and synovial fluid samples.</p>
<div id="plasma" class="section level3">
<h3>Plasma</h3>
<pre class="r"><code># Proteomics
plasma_proteomics_Znorm &lt;- t(apply(plasma_proteomics_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_proteomics_Znorm) &lt;- colnames(plasma_proteomics_data)

# miRNA
plasma_miRNA_Znorm &lt;- t(apply(plasma_miRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_miRNA_Znorm) &lt;- colnames(plasma_miRNA_data)


# lncRNA
plasma_lncRNA_Znorm &lt;- t(apply(plasma_lncRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_lncRNA_Znorm) &lt;- colnames(plasma_lncRNA_data)


# snRNA
plasma_snRNA_Znorm &lt;- t(apply(plasma_snRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_snRNA_Znorm) &lt;- colnames(plasma_snRNA_data)</code></pre>
<p>Concatenate plasma data together:</p>
<pre class="r"><code>plasma_data &lt;- rbind(plasma_proteomics_Znorm, plasma_miRNA_Znorm, plasma_lncRNA_Znorm, plasma_snRNA_Znorm)</code></pre>
</div>
<div id="synovial-fluid" class="section level3">
<h3>Synovial fluid</h3>
<pre class="r"><code># Proteomics
sf_proteomics_Znorm &lt;- t(apply(sf_proteomics_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_proteomics_Znorm) &lt;- colnames(sf_proteomics_data)

# miRNA
sf_miRNA_Znorm &lt;- t(apply(sf_miRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_miRNA_Znorm) &lt;- colnames(sf_miRNA_data)


# lncRNA
sf_lncRNA_Znorm &lt;- t(apply(sf_lncRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_lncRNA_Znorm) &lt;- colnames(sf_lncRNA_data)


# snRNA
sf_snRNA_Znorm &lt;- t(apply(sf_snRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_snRNA_Znorm) &lt;- colnames(sf_snRNA_data)</code></pre>
<p>Concatenate sf data together:</p>
<pre class="r"><code>sf_data &lt;- rbind(sf_proteomics_Znorm, sf_miRNA_Znorm, sf_lncRNA_Znorm, sf_snRNA_Znorm)</code></pre>
</div>
</div>
<div id="collapse-replicates-and-re-order-by-time"
class="section level2">
<h2>Collapse replicates and re-order by time</h2>
<p>The two clustering algorithms employed in this analysis do not
recognise multiple replicates in the same dataframe. Before clustering
the average z-score for each time point was calculated and the data was
re-ordered by time.</p>
<p>Plasma:</p>
<pre class="r"><code>plasma_time_df &lt;- data.frame(&quot;Day_0&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;0&quot;], na.rm=TRUE),
         &quot;Day_10&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;10&quot;], na.rm=TRUE),
         &quot;Day_35&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;35&quot;], na.rm=TRUE),
         &quot;Day_42&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;42&quot;], na.rm=TRUE),
         &quot;Day_49&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;49&quot;], na.rm=TRUE),
         &quot;Day_56&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;56&quot;], na.rm=TRUE),
         &quot;Day_63&quot; = rowMeans(plasma_data[,plasma_pheno$Day == &quot;63&quot;], na.rm=TRUE))


# Despite the filtering some time points will still end up with 0 samples for a particular feature
# These need to be removed
plasma_time_df &lt;- na.omit(plasma_time_df)</code></pre>
<p>Synovial fluid:</p>
<pre class="r"><code>sf_time_df &lt;- data.frame(&quot;Day_0&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;0&quot;], na.rm=TRUE),
         &quot;Day_10&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;10&quot;], na.rm=TRUE),
         &quot;Day_35&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;35&quot;], na.rm=TRUE),
         &quot;Day_42&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;42&quot;], na.rm=TRUE),
         &quot;Day_49&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;49&quot;], na.rm=TRUE),
         &quot;Day_56&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;56&quot;], na.rm=TRUE),
         &quot;Day_63&quot; = rowMeans(sf_data[,sf_pheno$Day == &quot;63&quot;], na.rm=TRUE))

# Despite the filtering some time points will still end up with 0 samples for a particular feature
# These need to be removed
sf_time_df &lt;- na.omit(sf_time_df)</code></pre>
</div>
</div>
<div id="mfuzz" class="section level1">
<h1>Mfuzz</h1>
<div id="plasma-1" class="section level2">
<h2>Plasma</h2>
<p>Create an <code>ExpressionSet</code> object containing the data:</p>
<pre class="r"><code>expData &lt;- ExpressionSet(as.matrix(plasma_time_df))</code></pre>
<p>Estimate the fuzzifier to designate how ‘fuzzy’ the clusters are. A
fuzzifier of 1 is almost equivalent to hard clustering.</p>
<pre class="r"><code>m1 &lt;- mestimate(expData)</code></pre>
<p>The <code>Dmin()</code> function calculates the minimum centroid
distance between clusters for a series of cluster numbers. We can use
this to determine the optimum number of clusters to use. After a certain
number of clusters the separation between clusters becomes minimal.
Visual inspection of the plot shows an ‘elbow’ to pick the appropriate k
number.</p>
<pre class="r"><code>Dmin(expData, m=m1, crange=seq(2,22,1), repeats=3, visu=TRUE)</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code> [1] 8.520583e-01 6.240207e-01 2.112929e-02 6.720354e-03 3.002782e-03
 [6] 2.024252e-03 5.406994e-04 3.146921e-04 1.518910e-04 7.297920e-05
[11] 8.196159e-05 4.936843e-05 2.063865e-05 2.496681e-04 1.247442e-04
[16] 6.570772e-05 5.193325e-05 1.207223e-04 1.046694e-04 1.729926e-04
[21] 1.834232e-04</code></pre>
<p>Visual inspection of the plot suggests &lt;5 clusters are
required.</p>
<pre class="r"><code>clust &lt;- 4</code></pre>
<pre class="r"><code># Carry out clustering using the calculated cluster number and fuzziness parameters
set.seed(5)
plasma_mfuzz_cl &lt;- mfuzz(expData,c=clust,m=m1)</code></pre>
<pre class="r"><code>cor(t(plasma_mfuzz_cl[[1]]))</code></pre>
<pre><code>           1          2          3          4
1  1.0000000 -0.3972422 -0.4712950 -0.4163115
2 -0.3972422  1.0000000  0.9794942  0.7908078
3 -0.4712950  0.9794942  1.0000000  0.8864916
4 -0.4163115  0.7908078  0.8864916  1.0000000</code></pre>
<p>The per-cluster membership can be plotted with the in-built Mfuzz
function <code>mfuzz.plot</code>.</p>
<pre class="r"><code>mfuzz.plot(eset = expData, cl = plasma_mfuzz_cl, mfrow = c(2,2))</code></pre>
<p>However, this doesn’t give the same level of control over the
aesthetics, so ggplot2 was used instead.</p>
<p>The centroids for each cluster were plotted on one graph along with
the per-cluster membership scores in separate graphs. Each cluster was
assigned a colour. For the per-cluster membership the opacity of each
feature was weighted by its membership score. The code was adapted from
the following tutorial:</p>
<p><a
href="https://2-bitbio.com/2017/10/clustering-rnaseq-data-using-k-means.html"
class="uri">https://2-bitbio.com/2017/10/clustering-rnaseq-data-using-k-means.html</a></p>
<p>To prepare the centroid data and plot:</p>
<pre class="r"><code># Convert centroid data into long dataframe
plasma_centroids &lt;- plasma_mfuzz_cl$centers
plasma_centroids_df &lt;- data.frame(plasma_centroids)
plasma_centroids_df$cluster &lt;- row.names(plasma_centroids_df)

# Convert dataframe into long format
plasma_centroids_long &lt;- tidyr::pivot_longer(plasma_centroids_df, names_to = &quot;sample&quot;, values_to = &quot;value&quot;, 1:length(colnames(plasma_centroids)))

# Remove &quot;Day_&quot; prefix from sample names
plasma_centroids_long$sample &lt;- gsub(&quot;Day_&quot;, &quot;&quot;, plasma_centroids_long$sample)</code></pre>
<pre class="r"><code># Centroid plot

plasma_mfuzz_centroids_plot &lt;- ggplot(plasma_centroids_long, aes(x=as.numeric(sample),y=value, group=cluster, colour=as.factor(cluster))) + 
  geom_line(lwd=1) +
  theme_cowplot() +
  scale_colour_brewer(palette = &quot;Set1&quot;) +
  xlab(&quot;Day&quot;) +
  ylab(&quot;Expression&quot;) +
  labs(title= &quot;Plasma EV fuzzy c-means expression&quot;,color = &quot;Cluster&quot;) </code></pre>
<p>To prepare the per-cluster feature membership data and plot:</p>
<pre class="r fold-hide"><code># Create copy of dataframe containing average z-score per time
plasma_membership_df &lt;- data.frame(plasma_time_df)

# Add row with feature names
plasma_membership_df$Feature &lt;- row.names(plasma_membership_df)

# Bind the cluster assignment 
plasma_membership_df$cluster &lt;- plasma_mfuzz_cl$cluster

# Fetch the membership for each feature in its top scoring cluster
plasma_membership_df$membership &lt;- sapply(1:length(plasma_membership_df$cluster),function(row){
  clust &lt;- plasma_membership_df$cluster[row]
  plasma_mfuzz_cl$membership[row,clust]
})


# Convert membership dataframe into long format
plasma_cluster_plot_df &lt;- plasma_membership_df %&gt;%
  dplyr::select(.,1:length(colnames(plasma_centroids)), membership, Feature, cluster) %&gt;%
  tidyr::gather(.,&quot;sample&quot;,&#39;value&#39;,1:length(colnames(plasma_centroids)))


# Remove &quot;Day_&quot; prefix from &quot;sample&quot; column
plasma_cluster_plot_df$sample &lt;- gsub(&quot;Day_&quot;, &quot;&quot;, plasma_cluster_plot_df$sample)


# Order the dataframe by membership score
plasma_cluster_plot_df &lt;- plasma_cluster_plot_df[order(plasma_cluster_plot_df$membership),]


# Convert the features into an ordered factor using forcats
plasma_cluster_plot_df$Feature = forcats::fct_inorder(plasma_cluster_plot_df$Feature)</code></pre>
<pre class="r fold-hide"><code># Per-cluster membership plot

plasma_mfuzz_plot_membership &lt;- ggplot(plasma_cluster_plot_df, aes(x=as.numeric(sample),y=value)) + 
        # Facet-wrap by cluster
        facet_wrap(~cluster, scales=&quot;free_y&quot;) +
        # Colour each panel by cluster
        # The oppacity of each feature is determined by membership
        geom_line(aes(colour=as.factor(cluster), alpha=membership, group =Feature), lwd=1) +
        scale_colour_brewer(palette = &quot;Set1&quot;) +
        # Add the centroids as a black line
        geom_line(data=plasma_centroids_long, aes(x=as.numeric(sample), y=value, group=cluster), lwd=1.2, color=&quot;black&quot;,inherit.aes=FALSE) +
        xlab(&quot;Day&quot;) +
        ylab(&quot;Expression&quot;) +
        labs(title=&quot;Per-cluster membership&quot;) + 
        theme_cowplot() + 
        theme(legend.position = &quot;none&quot;)</code></pre>
<p>To visualise the results together:</p>
<pre class="r"><code>ggarrange(plasma_mfuzz_centroids_plot, plasma_mfuzz_plot_membership,                        
          nrow = 1, 
          ncol = 2,
          labels = c(&quot;A&quot;, &quot;B&quot;))</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-24-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Cluster 4 appears to display the most interesting pattern over time,
with an initial dip after OA induction followed by increased expression
over the time course. As cluster 2 &amp; 3 have a high overlap they are
likely redundant clusters. Cluster 1 has a distinct profile, though as
it fluctuates from high to low its difficult to speculate how it
translates to OA progression.</p>
<p>For downstream analysis the membership scores for each gene by
cluster were extracted:</p>
<pre class="r"><code># Extracts membership values 
plasma_acore &lt;- acore(expData, plasma_mfuzz_cl, min.acore=0)


# Create a combined dataframe of the combined clusters
plasma_acore_list &lt;- do.call(rbind, lapply(seq_along(plasma_acore), function(i){ data.frame(CLUSTER=i, plasma_acore[[i]])}))</code></pre>
<p>It is interesting that eca-miR-186<span class="citation">(Li et al.
2021)</span>, F7C1Y3<span class="citation">(Chou et al. 2020)</span> and
eca-miR-126-5p were among the members of cluster 4.</p>
</div>
<div id="synovial-fluid-1" class="section level2">
<h2>Synovial fluid</h2>
<p>The same steps were repeated but for the synovial fluid data instead.
Create an <code>ExpressionSet</code> object containing the data:</p>
<pre class="r"><code>expData &lt;- ExpressionSet(as.matrix(sf_time_df))</code></pre>
<p>Estimate the fuzzifier to designate how ‘fuzzy’ the clusters are. A
fuzzifier of 1 is almost equivalent to hard clustering.</p>
<pre class="r"><code>m1 &lt;- mestimate(expData)</code></pre>
<p>Estimate number of clusters required:</p>
<pre class="r"><code>Dmin(expData, m=m1, crange=seq(2,22,1), repeats=3, visu=TRUE)</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre><code> [1] 1.115852599 0.535072006 0.301359314 0.217062356 0.024929588 0.021075170
 [7] 0.029505365 0.011466759 0.013917440 0.012368646 0.016592531 0.007800507
[13] 0.010526264 0.001905725 0.009248844 0.005652783 0.009083171 0.004543876
[19] 0.006119594 0.003049429 0.004465958</code></pre>
<p>Visual inspection of the plot suggests &lt;7 clusters are
required.</p>
<pre class="r"><code>clust &lt;- 6</code></pre>
<pre class="r"><code># Carry out clustering using the calculated cluster number and fuzziness parameters
set.seed(5)
sf_mfuzz_cl &lt;- mfuzz(expData,c=clust,m=m1)</code></pre>
<pre class="r"><code>cor(t(sf_mfuzz_cl[[1]]))</code></pre>
<pre><code>           1          2          3          4          5          6
1  1.0000000 -0.7555527 -0.9726895  0.5737932 -0.2234023 -0.7644797
2 -0.7555527  1.0000000  0.6504193 -0.3692796 -0.2389322  0.9994863
3 -0.9726895  0.6504193  1.0000000 -0.5409932  0.3312384  0.6653548
4  0.5737932 -0.3692796 -0.5409932  1.0000000 -0.7196810 -0.3685807
5 -0.2234023 -0.2389322  0.3312384 -0.7196810  1.0000000 -0.2307242
6 -0.7644797  0.9994863  0.6653548 -0.3685807 -0.2307242  1.0000000</code></pre>
<p>Results suggest 5&amp;6 may be redundant clusters due to correlation
between their centroids.</p>
<p>Plot centroid expression over time and per-cluster membership scores
as before. First prepare the centroid data and plot:</p>
<pre class="r"><code># Convert centroid data into long dataframe
sf_centroids &lt;- sf_mfuzz_cl$centers
sf_centroids_df &lt;- data.frame(sf_centroids)
sf_centroids_df$cluster &lt;- row.names(sf_centroids_df)

# Convert dataframe into long format
sf_centroids_long &lt;- tidyr::pivot_longer(sf_centroids_df, names_to = &quot;sample&quot;, values_to = &quot;value&quot;, 1:length(colnames(sf_centroids)))

# Remove &quot;Day_&quot; prefix from sample names
sf_centroids_long$sample &lt;- gsub(&quot;Day_&quot;, &quot;&quot;, sf_centroids_long$sample)</code></pre>
<pre class="r"><code># Centroid plot

sf_mfuzz_centroids_plot &lt;- ggplot(sf_centroids_long, aes(x=as.numeric(sample),y=value, group=cluster, colour=as.factor(cluster))) + 
  geom_line(lwd=1) +
  theme_cowplot() +
  #scale_colour_brewer(palette = &quot;Set1&quot;) +
  scale_color_met_d(&quot;Hiroshige&quot;) +
  xlab(&quot;Day&quot;) +
  ylab(&quot;Expression&quot;) +
  labs(title= &quot;SF EV fuzzy c-means expression&quot;,color = &quot;Cluster&quot;) </code></pre>
<p>Next prepare the per cluster feature membership data and plot:</p>
<pre class="r"><code># Create copy of dataframe containing average z-score per time
sf_membership_df &lt;- data.frame(sf_time_df)

# Add row with feature names
sf_membership_df$Feature &lt;- row.names(sf_membership_df)

# Bind the cluster assignment 
sf_membership_df$cluster &lt;- sf_mfuzz_cl$cluster

# Fetch the membership for each feature in its top scoring cluster
sf_membership_df$membership &lt;- sapply(1:length(sf_membership_df$cluster),function(row){
  clust &lt;- sf_membership_df$cluster[row]
  sf_mfuzz_cl$membership[row,clust]
})


# Convert membership dataframe into long format
sf_cluster_plot_df &lt;- sf_membership_df %&gt;%
  dplyr::select(.,1:length(colnames(sf_centroids)), membership, Feature, cluster) %&gt;%
  tidyr::gather(.,&quot;sample&quot;,&#39;value&#39;,1:length(colnames(sf_centroids)))


# Remove &quot;Day_&quot; prefix from &quot;sample&quot; column
sf_cluster_plot_df$sample &lt;- gsub(&quot;Day_&quot;, &quot;&quot;, sf_cluster_plot_df$sample)


# Order the dataframe by membership score
sf_cluster_plot_df &lt;- sf_cluster_plot_df[order(sf_cluster_plot_df$membership),]


# Convert the features into an ordered factor using forcats
sf_cluster_plot_df$Feature = forcats::fct_inorder(sf_cluster_plot_df$Feature)</code></pre>
<pre class="r"><code># Per-cluster membership plot

sf_mfuzz_plot_membership &lt;- ggplot(sf_cluster_plot_df, aes(x=as.numeric(sample),y=value)) + 
        # Facet-wrap by cluster
        facet_wrap(~cluster, scales=&quot;free_y&quot;) +
        # Colour each panel by cluster
        # The oppacity of each feature is determined by membership
        geom_line(aes(colour=as.factor(cluster), alpha=membership, group =Feature), lwd=1) +
        scale_color_met_d(&quot;Hiroshige&quot;) +
        # Add the centroids as a black line
        geom_line(data=sf_centroids_long, aes(x=as.numeric(sample), y=value, group=cluster), lwd=1.2, color=&quot;black&quot;,inherit.aes=FALSE) +
        xlab(&quot;Day&quot;) +
        ylab(&quot;Expression&quot;) +
        labs(title=&quot;Per-cluster membership&quot;) + 
        theme_cowplot() + 
        theme(legend.position = &quot;none&quot;)</code></pre>
<pre class="r"><code>ggarrange(sf_mfuzz_centroids_plot, sf_mfuzz_plot_membership,                        
          nrow = 1, 
          ncol = 2,
          labels = c(&quot;A&quot;, &quot;B&quot;))</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-36-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Extract membership scores for each gene by cluster:</p>
<pre class="r"><code># Extracts membership values 
sf_acore &lt;- acore(expData, sf_mfuzz_cl, min.acore=0)


# Create a combined dataframe of the combined clusters
sf_acore_list &lt;- do.call(rbind, lapply(seq_along(sf_acore), function(i){ data.frame(CLUSTER=i, sf_acore[[i]])}))</code></pre>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>Combine plasma and synovial fluid soft clustering results
together:</p>
<pre class="r"><code>ggarrange(ggarrange(plasma_mfuzz_centroids_plot, plasma_mfuzz_plot_membership,                        
          nrow = 1, 
          ncol = 2,
          labels = c(&quot;A&quot;, &quot;B&quot;)),
ggarrange(sf_mfuzz_centroids_plot, sf_mfuzz_plot_membership,                        
          nrow = 1, 
          ncol = 2,
          labels = c(&quot;C&quot;, &quot;D&quot;)),
          nrow = 2,
          ncol = 1)</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-38-1.png" width="1008" style="display: block; margin: auto;" /></p>
<p>Overall both datasets have clusters that demonstrate interesting
time-dependent behaviour. For the synovial fluid data clusters 4 &amp; 5
have two inverse profiles at day 10. This may be part of the initial
immune response to the injury/tissue insult. This was also seen with the
differential expression analysis (limma).</p>
</div>
</div>
<div id="dtwclust" class="section level1">
<h1>dtwclust</h1>
<p>To make the analysis more robust multiple clustering methods were
compared. Both <code>Mfuzz</code> and <code>dtwclust</code> are
time-series clustering packages. For <code>dtwclust</code> k-medoid
clustering (PAM) was employed. K-medoid is a form of partitional
clustering. Unlike fuzzy c-means, it is more sensitive to noise in the
data, which is why variance filtering was applied earlier.</p>
<div id="plasma-2" class="section level2">
<h2>Plasma</h2>
<pre class="r"><code>plasma_dtwclust_cl &lt;- tsclust(na.omit(plasma_time_df), type = &quot;partitional&quot;, k = 4L, 
              distance = &quot;dtw&quot;, centroid = &quot;pam&quot;, 
              seed = 3247L, trace = TRUE,
              args = tsclust_args(dist = list(window.size = 20L)))</code></pre>
<pre><code>
    Precomputing distance matrix...

Iteration 1: Changes / Distsum = 173 / 557.6487
Iteration 2: Changes / Distsum = 75 / 472.9479
Iteration 3: Changes / Distsum = 0 / 472.9479

    Elapsed time is 10.06 seconds.</code></pre>
<pre class="r"><code># Prepare data for centroid plot

## Convert list object into matrix
plasma_dtw_centroids &lt;- do.call(rbind, plasma_dtwclust_cl@centroids)

## Rename columns in matrix
colnames(plasma_dtw_centroids) &lt;- colnames(plasma_time_df)

## Convert to dataframe
plasma_dtw_centroids_df &lt;- data.frame(plasma_dtw_centroids)

## Add column for cluster
plasma_dtw_centroids_df$cluster &lt;- row.names(plasma_dtw_centroids_df)

## Convert dataframe into long format
plasma_dtw_centroids_long &lt;- tidyr::pivot_longer(plasma_dtw_centroids_df, names_to = &quot;sample&quot;, values_to = &quot;value&quot;, 1:length(colnames(plasma_dtw_centroids)))

## Remove &quot;Day_&quot; prefix from sample names
plasma_dtw_centroids_long$sample &lt;- gsub(&quot;Day_&quot;, &quot;&quot;, plasma_centroids_long$sample)



# Centroid plot

plasma_dtw_centroids_plot &lt;- ggplot(plasma_dtw_centroids_long, aes(x=as.numeric(sample),y=value, group=cluster, colour=as.factor(cluster))) + 
  geom_line(lwd=1) +
  theme_cowplot() +
  scale_colour_brewer(palette = &quot;Set1&quot;) +
  xlab(&quot;Day&quot;) +
  ylab(&quot;Expression&quot;) +
  labs(title= &quot;Plasma K-medoid expression&quot;, color = &quot;Cluster&quot;)</code></pre>
</div>
<div id="synovial-fluid-2" class="section level2">
<h2>Synovial fluid</h2>
<pre class="r"><code>sf_dtwclust_cl &lt;- tsclust(na.omit(sf_time_df), type = &quot;partitional&quot;, k = 6L, 
                              distance = &quot;dtw&quot;, centroid = &quot;pam&quot;, 
                              seed = 3247L, trace = TRUE,
                              args = tsclust_args(dist = list(window.size = 20L)))</code></pre>
<pre><code>
    Precomputing distance matrix...

Iteration 1: Changes / Distsum = 337 / 1104.139
Iteration 2: Changes / Distsum = 142 / 914.9503
Iteration 3: Changes / Distsum = 11 / 908.5555
Iteration 4: Changes / Distsum = 0 / 908.5555

    Elapsed time is 37.94 seconds.</code></pre>
<pre class="r"><code># Prepare data for centroid plot

## Convert list object into matrix
sf_dtw_centroids &lt;- do.call(rbind, sf_dtwclust_cl@centroids)

## Rename columns in matrix
colnames(sf_dtw_centroids) &lt;- colnames(sf_time_df)

## Convert to dataframe
sf_dtw_centroids_df &lt;- data.frame(sf_dtw_centroids)

## Add column for cluster
sf_dtw_centroids_df$cluster &lt;- row.names(sf_dtw_centroids_df)

## Convert dataframe into long format
sf_dtw_centroids_long &lt;- tidyr::pivot_longer(sf_dtw_centroids_df, names_to = &quot;sample&quot;, values_to = &quot;value&quot;, 1:length(colnames(sf_dtw_centroids)))

## Remove &quot;Day_&quot; prefix from sample names
sf_dtw_centroids_long$sample &lt;- gsub(&quot;Day_&quot;, &quot;&quot;, sf_centroids_long$sample)



# Centroid plot

sf_dtw_centroids_plot &lt;- ggplot(sf_dtw_centroids_long, aes(x=as.numeric(sample),y=value, group=cluster, colour=as.factor(cluster))) + 
  geom_line(lwd=1) +
  theme_cowplot() +
  scale_color_met_d(&quot;Hiroshige&quot;) +
  xlab(&quot;Day&quot;) +
  ylab(&quot;Expression&quot;) +
  labs(title= &quot;SF K-medoid expression&quot;, color = &quot;Cluster&quot;)</code></pre>
</div>
<div id="summary-1" class="section level2">
<h2>Summary</h2>
<pre class="r"><code>ggarrange(plasma_dtw_centroids_plot, sf_dtw_centroids_plot,
          nrow = 1,
          labels = c(&quot;A&quot;, &quot;B&quot;))</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-43-1.png" width="1056" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="compare-clustering-methods" class="section level1">
<h1>Compare clustering methods</h1>
<div id="plasma-3" class="section level2">
<h2>Plasma</h2>
<p>Extract Mfuzz and dtwclust cluster assignments:</p>
<pre class="r"><code># dtwclust
plasma_dtw_cl &lt;- plasma_dtwclust_cl@cluster 
names(plasma_dtw_cl) &lt;- row.names(plasma_time_df)


# Mfuzz
plasma_mfuzz_clusters &lt;- plasma_mfuzz_cl$cluster


# Check names are the same order
names(plasma_dtw_cl) == names(plasma_mfuzz_clusters)</code></pre>
<pre><code>  [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [16] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [31] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [46] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [61] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [76] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [91] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[106] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[121] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[136] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[151] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[166] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code># Add to cluster labels:
plasma_dtw_cl_matched &lt;- paste0(&#39;K-&#39;,plasma_dtw_cl)
plasma_mfuzz_clusters &lt;- paste0(&#39;FCM-&#39;, plasma_mfuzz_clusters)

# Calculate the overlap
OL&lt;- overlapTable(plasma_dtw_cl_matched, plasma_mfuzz_clusters)

# Get rid of 0 values...
OL$pTable[OL$pTable == 0] &lt;- 2e-300</code></pre>
<pre class="r"><code># Extract overlap counts
overlap_counts &lt;- signif(OL$countTable, 2)

# Create matrix of labels for pheatmap
text_labels &lt;- paste(signif(OL$countTable, 2), &quot;\n(&quot;,
                      signif(OL$pTable, 1), &quot;)&quot;, sep= &quot;&quot;)
dim(text_labels) &lt;-  dim(OL$countTable)

# Create heatmap of overlap 
plasma_cl_OL_plot &lt;- pheatmap(overlap_counts,
               display_numbers = text_labels,
               cluster_rows=FALSE,
               cluster_cols=FALSE,
               border_color = NA,
               #color = mako(50),
               color = hcl.colors(50, &quot;BluYl&quot;),
               fontsize_number = 14,
               fontsize_row = 15,
               fontsize_col = 15,
               number_color = &quot;black&quot;,
               legend = FALSE)</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-47-1.png" width="432" style="display: block; margin: auto;" /></p>
<p>Note, to access the heatmap for functions such as ggarrange, extract
the 4th element, e.g. <code>plasma_cl_OL_plot[[4]]</code>.</p>
<p>Fuzzy c-means cluster 3 doesn’t appear to have a corresponding
k-medoids cluster. This appears to stem from redundancy between cluster
2&amp;3. It is probably worth reducing the number of clusters to 3 or
not including cluster 3 in any downstream analysis.</p>
</div>
<div id="synovial-fluid-3" class="section level2">
<h2>Synovial fluid</h2>
<p>Extract Mfuzz and dtwclust cluster assignments:</p>
<pre class="r"><code># dtwclust
sf_dtw_cl &lt;- sf_dtwclust_cl@cluster 
names(sf_dtw_cl) &lt;- row.names(sf_time_df)


# Mfuzz
sf_mfuzz_clusters &lt;- sf_mfuzz_cl$cluster


# Check names are the same order
names(sf_dtw_cl) == names(sf_mfuzz_clusters)</code></pre>
<pre><code>  [1] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [16] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [31] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [46] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [61] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [76] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
 [91] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[106] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[121] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[136] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[151] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[166] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[181] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[196] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[211] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[226] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[241] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[256] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[271] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[286] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[301] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[316] TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE TRUE
[331] TRUE TRUE TRUE TRUE TRUE TRUE TRUE</code></pre>
<pre class="r"><code># Add to cluster labels:
sf_dtw_cl_matched &lt;- paste0(&#39;K-&#39;,sf_dtw_cl)
sf_mfuzz_clusters &lt;- paste0(&#39;FCM-&#39;, sf_mfuzz_clusters)

# Calculate the overlap
OL&lt;- overlapTable(sf_dtw_cl_matched, sf_mfuzz_clusters)

# Get rid of 0 values...
OL$pTable[OL$pTable == 0] &lt;- 2e-300</code></pre>
<pre class="r"><code># Extract overlap counts
overlap_counts &lt;- signif(OL$countTable, 2)

# Create matrix of labels for pheatmap
text_labels &lt;- paste(signif(OL$countTable, 2), &quot;\n(&quot;,
                     signif(OL$pTable, 1), &quot;)&quot;, sep= &quot;&quot;)
dim(text_labels) &lt;-  dim(OL$countTable)

# Create heatmap of overlap 
sf_cl_OL_plot &lt;- pheatmap(overlap_counts,
                              display_numbers = text_labels,
                              cluster_rows=FALSE,
                              cluster_cols=FALSE,
                              border_color = NA,
                              #color = mako(50),
                              color = hcl.colors(50, &quot;BluYl&quot;),
                              fontsize_number = 14,
                              fontsize_row = 15,
                              fontsize_col = 15,
                              number_color = &quot;black&quot;,
                              legend = FALSE)</code></pre>
<p><img src="figure/07_Clustering.Rmd/unnamed-chunk-51-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>The results demonstrate some overlap between the clustering
algorithms, especially for cluster 5. Fuzzy c-means 3 and k-medoids 2
also demonstrate strong overlap.</p>
<hr />
</div>
</div>
<div id="write-outputs" class="section level1">
<h1>Write outputs</h1>
<p>The mfuzz fuzzy c-means analysis was taken forward for functional
enrichment as its was reasonable agreement with the dtwclust for the
most relevant clusters. Additionally it includes a membership score
which is useful for functional enrichment techniques that need a ranked
list (though this would be possible to generate for the dtwclust results
too by correlating the features with the centroids and applying a
cut-off if this was of interest).</p>
<pre class="r"><code>saveRDS(plasma_acore_list, &quot;./data/mfuzz_cluster_plasma_data.rds&quot;)
saveRDS(sf_acore_list, &quot;./data/mfuzz_cluster_sf_data.rds&quot;)</code></pre>
<!-- Below is some code that will be hidden in the outputs to save figures. -->
</div>
<div id="references" class="section level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-chou2020" class="csl-entry">
Chou, Ching-Heng, Vaibhav Jain, Jason Gibson, David E. Attarian, Collin
A. Haraden, Christopher B. Yohn, Remi-Martin Laberge, Simon Gregory, and
Virginia B. Kraus. 2020. <span>“Synovial Cell Cross-Talk with Cartilage
Plays a Major Role in the Pathogenesis of Osteoarthritis.”</span>
<em>Scientific Reports</em> 10 (July): 10868. <a
href="https://doi.org/10.1038/s41598-020-67730-y">https://doi.org/10.1038/s41598-020-67730-y</a>.
</div>
<div id="ref-li2021" class="csl-entry">
Li, Qing, Mingjie Wu, Guofang Fang, Kuangwen Li, Wengang Cui, Liang Li,
Xia Li, Junsheng Wang, and Yanhong Cang. 2021. <span>“MicroRNA-186-5p
Downregulation Inhibits Osteoarthritis Development by Targeting
Mapk1.”</span> <em>Molecular Medicine Reports</em> 23 (4): 253. <a
href="https://doi.org/10.3892/mmr.2021.11892">https://doi.org/10.3892/mmr.2021.11892</a>.
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

Random number generation:
 RNG:     L&#39;Ecuyer-CMRG 
 Normal:  Inversion 
 Sample:  Rejection 
 
locale:
[1] LC_COLLATE=English_United Kingdom.utf8 
[2] LC_CTYPE=English_United Kingdom.utf8   
[3] LC_MONETARY=English_United Kingdom.utf8
[4] LC_NUMERIC=C                           
[5] LC_TIME=English_United Kingdom.utf8    

attached base packages:
[1] tcltk     stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] pheatmap_1.0.12       WGCNA_1.71            fastcluster_1.2.3    
 [4] dynamicTreeCut_1.63-1 MetBrewer_0.2.0       ggpubr_0.4.0         
 [7] cowplot_1.1.1         dtwclust_5.5.11       dtw_1.23-1           
[10] proxy_0.4-27          Mfuzz_2.56.0          DynDoc_1.74.0        
[13] widgetTools_1.74.0    e1071_1.7-11          Biobase_2.56.0       
[16] BiocGenerics_0.42.0   forcats_0.5.1         stringr_1.4.0        
[19] dplyr_1.0.9           purrr_0.3.4           readr_2.1.2          
[22] tidyr_1.2.0           tibble_3.1.7          ggplot2_3.3.6        
[25] tidyverse_1.3.1      

loaded via a namespace (and not attached):
  [1] readxl_1.4.0           backports_1.4.1        Hmisc_4.7-0           
  [4] workflowr_1.7.0        plyr_1.8.7             splines_4.2.0         
  [7] GenomeInfoDb_1.32.2    digest_0.6.29          foreach_1.5.2         
 [10] htmltools_0.5.2        GO.db_3.15.0           fansi_1.0.3           
 [13] checkmate_2.1.0        magrittr_2.0.3         memoise_2.0.1         
 [16] cluster_2.1.3          doParallel_1.0.17      tzdb_0.3.0            
 [19] Biostrings_2.64.0      modelr_0.1.8           RcppParallel_5.1.5    
 [22] matrixStats_0.62.0     jpeg_0.1-9             colorspace_2.0-3      
 [25] blob_1.2.3             rvest_1.0.2            ggrepel_0.9.1         
 [28] haven_2.5.0            xfun_0.31              crayon_1.5.1          
 [31] RCurl_1.98-1.7         jsonlite_1.8.0         impute_1.70.0         
 [34] survival_3.3-1         iterators_1.0.14       glue_1.6.2            
 [37] gtable_0.3.0           zlibbioc_1.42.0        XVector_0.36.0        
 [40] car_3.1-0              abind_1.4-5            scales_1.2.0          
 [43] DBI_1.1.2              rstatix_0.7.0          Rcpp_1.0.8.3          
 [46] htmlTable_2.4.0        xtable_1.8-4           clue_0.3-61           
 [49] foreign_0.8-82         bit_4.0.4              preprocessCore_1.58.0 
 [52] Formula_1.2-4          stats4_4.2.0           htmlwidgets_1.5.4     
 [55] httr_1.4.3             RColorBrewer_1.1-3     modeltools_0.2-23     
 [58] ellipsis_0.3.2         farver_2.1.0           pkgconfig_2.0.3       
 [61] nnet_7.3-17            sass_0.4.1             dbplyr_2.2.0          
 [64] utf8_1.2.2             labeling_0.4.2         tidyselect_1.1.2      
 [67] rlang_1.0.2            tkWidgets_1.74.0       reshape2_1.4.4        
 [70] later_1.3.0            AnnotationDbi_1.58.0   munsell_0.5.0         
 [73] cellranger_1.1.0       tools_4.2.0            cachem_1.0.6          
 [76] cli_3.3.0              generics_0.1.2         RSQLite_2.2.14        
 [79] broom_0.8.0            evaluate_0.15          fastmap_1.1.0         
 [82] yaml_2.3.5             knitr_1.39             bit64_4.0.5           
 [85] fs_1.5.2               KEGGREST_1.36.2        whisker_0.4           
 [88] mime_0.12              xml2_1.3.3             flexclust_1.4-1       
 [91] compiler_4.2.0         rstudioapi_0.13        png_0.1-7             
 [94] ggsignif_0.6.3         reprex_2.0.1           bslib_0.3.1           
 [97] stringi_1.7.6          highr_0.9              RSpectra_0.16-1       
[100] lattice_0.20-45        Matrix_1.5-1           shinyjs_2.1.0         
[103] vctrs_0.4.1            pillar_1.7.0           lifecycle_1.0.1       
[106] jquerylib_0.1.4        data.table_1.14.2      bitops_1.0-7          
[109] httpuv_1.6.5           R6_2.5.1               latticeExtra_0.6-29   
[112] promises_1.2.0.1       gridExtra_2.3          IRanges_2.30.0        
[115] codetools_0.2-18       assertthat_0.2.1       rprojroot_2.0.3       
[118] withr_2.5.0            S4Vectors_0.34.0       GenomeInfoDbData_1.2.8
[121] parallel_4.2.0         hms_1.1.1              rpart_4.1.16          
[124] grid_4.2.0             class_7.3-20           rmarkdown_2.14        
[127] carData_3.0-5          git2r_0.30.1           shiny_1.7.1           
[130] lubridate_1.8.0        base64enc_0.1-3       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
