---
title: "Multi-omics - clustering"
author: "Emily Johnson"
date: "2022-09-28"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Load libraries

```{r, message=FALSE, warning=FALSE}

source("./code/utility-functions.R") # load helper functions 
library(tidyverse)
library(Mfuzz)
library(dtwclust)
library(cowplot) # ggplot2 theme

```

# Load and process data

```{r}

# Plasma 
plasma_annotation <- readRDS("./data/plasma_proteomics_annotation.rds")
plasma_pheno <- readRDS("./data/plasma_pheno.rds") 
plasma_proteomics_data <- readRDS("./data/plasma_data_norm.rds")
plasma_miRNA_data <- readRDS("./data/plasma_miRNA_data.rds") 
plasma_lncRNA_data <- readRDS("./data/plasma_lncRNA_data.rds")
plasma_snRNA_data <- readRDS("./data/plasma_snRNA_data.rds")

# SF
sf_annotation <- readRDS("./data/sf_proteomics_annotation.rds")
sf_pheno <- readRDS("./data/sf_pheno.rds")
sf_proteomics_data <- readRDS("./data/sf_data_norm.rds")
sf_miRNA_data <- readRDS("./data/sf_miRNA_data.rds")
sf_lncRNA_data <- readRDS("./data/sf_lncRNA_data.rds")
sf_snRNA_data <- readRDS("./data/sf_snRNA_data.rds")


```

Convert 0s to NAs as previous analysis showed they were sampling zeroes rather than biological zeroes: 

```{r}

# Plasma 
plasma_miRNA_data[plasma_miRNA_data == 0] <- NA
plasma_lncRNA_data[plasma_lncRNA_data == 0] <- NA
plasma_snRNA_data[plasma_snRNA_data == 0] <- NA

# SF
sf_miRNA_data[sf_miRNA_data == 0] <- NA
sf_lncRNA_data[sf_lncRNA_data == 0] <- NA
sf_snRNA_data[sf_snRNA_data == 0] <- NA


```

Remove universal spike-ins:

```{r}

plasma_miRNA_data <- plasma_miRNA_data[-c(1:3),]
sf_miRNA_data <- sf_miRNA_data[-c(1:3),]

```


## Variance filtering

Before Z-score normalisation we filter the data to only include features with high variance. This is because protein/sequencing data is noisy. Noisy features can then be assigned to clusters even if their expression pattern isn’t representative of the cluster.

Plasma:

```{r}

## Create vectors of variance/mean value per protein
vars <- apply(plasma_proteomics_data, 1, var, na.rm=TRUE)
means <- apply(plasma_proteomics_data, 1, mean, na.rm=TRUE)

## Subset the data to only include proteins with a variance >1 and mean >2
plasma_proteomics_data <- plasma_proteomics_data[which(vars > 1 & means >2),]

## Remove any rows that contain >20% missing values
plasma_proteomics_data <- plasma_proteomics_data[which(rowMeans(is.na(plasma_proteomics_data)) < 0.2), ]

```


Synovial fluid: 

```{r}

## Create vectors of variance/mean value per protein
vars <- apply(sf_proteomics_data, 1, var, na.rm=TRUE)
means <- apply(sf_proteomics_data, 1, mean, na.rm=TRUE)

## Subset the data to only include proteins with a variance >1 and mean >2
sf_proteomics_data <- sf_proteomics_data[which(vars > 1 & means >2),]

## Remove any rows that contain >20% missing values
sf_proteomics_data <- sf_proteomics_data[which(rowMeans(is.na(sf_proteomics_data)) < 0.2), ]

```



## Z-score normalisation

### Plasma

```{r}

# Proteomics
plasma_proteomics_Znorm <- t(apply(plasma_proteomics_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_proteomics_Znorm) <- colnames(plasma_proteomics_data)

# miRNA
plasma_miRNA_Znorm <- t(apply(plasma_miRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_miRNA_Znorm) <- colnames(plasma_miRNA_data)


# lncRNA
plasma_lncRNA_Znorm <- t(apply(plasma_lncRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_lncRNA_Znorm) <- colnames(plasma_lncRNA_data)


# snRNA
plasma_snRNA_Znorm <- t(apply(plasma_snRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(plasma_snRNA_Znorm) <- colnames(plasma_snRNA_data)

```


Concatenate plasma data together:

```{r}

plasma_data <- rbind(plasma_proteomics_Znorm, plasma_miRNA_Znorm, plasma_lncRNA_Znorm, plasma_snRNA_Znorm)

```


### Synovial fluid

```{r}

# Proteomics
sf_proteomics_Znorm <- t(apply(sf_proteomics_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_proteomics_Znorm) <- colnames(sf_proteomics_data)

# miRNA
sf_miRNA_Znorm <- t(apply(sf_miRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_miRNA_Znorm) <- colnames(sf_miRNA_data)


# lncRNA
sf_lncRNA_Znorm <- t(apply(sf_lncRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_lncRNA_Znorm) <- colnames(sf_lncRNA_data)


# snRNA
sf_snRNA_Znorm <- t(apply(sf_snRNA_data, 1, function(x) scale(x, center = T, scale = T)))
colnames(sf_snRNA_Znorm) <- colnames(sf_snRNA_data)

```


Concatenate sf data together:
  
```{r}

sf_data <- rbind(sf_proteomics_Znorm, sf_miRNA_Znorm, sf_lncRNA_Znorm, sf_snRNA_Znorm)

```



## Collapse replicates and re-order by time

Plasma:

```{r}

plasma_time_df <- data.frame("Day_0" = rowMeans(plasma_data[,plasma_pheno$Day == "0"], na.rm=TRUE),
         "Day_10" = rowMeans(plasma_data[,plasma_pheno$Day == "10"], na.rm=TRUE),
         "Day_35" = rowMeans(plasma_data[,plasma_pheno$Day == "35"], na.rm=TRUE),
         "Day_42" = rowMeans(plasma_data[,plasma_pheno$Day == "42"], na.rm=TRUE),
         "Day_49" = rowMeans(plasma_data[,plasma_pheno$Day == "49"], na.rm=TRUE),
         "Day_56" = rowMeans(plasma_data[,plasma_pheno$Day == "56"], na.rm=TRUE),
         "Day_63" = rowMeans(plasma_data[,plasma_pheno$Day == "63"], na.rm=TRUE))


# Despite the filtering some time points will still end up with 0 samples for a particular feature
# These need to be removed
plasma_time_df <- na.omit(plasma_time_df)


```

Synovial fluid:
  
```{r}

sf_time_df <- data.frame("Day_0" = rowMeans(sf_data[,sf_pheno$Day == "0"], na.rm=TRUE),
         "Day_10" = rowMeans(sf_data[,sf_pheno$Day == "10"], na.rm=TRUE),
         "Day_35" = rowMeans(sf_data[,sf_pheno$Day == "35"], na.rm=TRUE),
         "Day_42" = rowMeans(sf_data[,sf_pheno$Day == "42"], na.rm=TRUE),
         "Day_49" = rowMeans(sf_data[,sf_pheno$Day == "49"], na.rm=TRUE),
         "Day_56" = rowMeans(sf_data[,sf_pheno$Day == "56"], na.rm=TRUE),
         "Day_63" = rowMeans(sf_data[,sf_pheno$Day == "63"], na.rm=TRUE))

# Despite the filtering some time points will still end up with 0 samples for a particular feature
# These need to be removed
sf_time_df <- na.omit(sf_time_df)

```


# Mfuzz

Create an `ExpressionSet` object containing the data:

```{r}

expData <- ExpressionSet(as.matrix(plasma_time_df))

```


Estimate the fuzzifier to designate how ‘fuzzy’ the clusters are. A fuzzifier of 1 is almost equivalent to hard clustering. 

```{r}

m1 <- mestimate(expData)

```


The `Dmin()` function calculates the minimum centroid distance between clusters for a series of cluster numbers. We can use this to determine the optimum number of clusters to use. After a certain number of clusters the separation between clusters becomes minimal. Visual inspection of the plot shows an ‘elbow’ to pick the appropriate k number.


```{r}

Dmin(expData, m=m1, crange=seq(2,22,1), repeats=3, visu=TRUE)

```

Visual inspection of the plot suggests <5 clusters are required. 

```{r}

clust <- 4

```

```{r}

# Carry out clustering using the calculated cluster number and fuzziness parameters
mfuzz_cl <- mfuzz(expData,c=clust,m=m1)

```

```{r}

mfuzz.plot(eset = expData, cl = mfuzz_cl, mfrow = c(2,2))

```


```{r}

time <- gsub("Day_", "", colnames(plasma_time_df))

```

```{r}

# Convert centroid data into long dataframe: 

fcm_centroids <- mfuzz_cl$centers
fcm_centroids_df <- data.frame(fcm_centroids)
fcm_centroids_df$cluster <- row.names(fcm_centroids_df)
centroids_long <- tidyr::pivot_longer(fcm_centroids_df, names_to = "sample", values_to = "value", 1:length(colnames(fcm_centroids)))


```

```{r}

ggplot(centroids_long, aes(x=sample,y=value, group=cluster, colour=as.factor(cluster))) + 
  geom_line(lwd=1) +
  theme_cowplot() +
  scale_colour_brewer(palette = "Set1") +
  xlab("Day") +
  ylab("Expression") +
  labs(title= "Cluster Expression by Time",color = "Cluster")

```


```{r}

#start with the input data
fcm_plotting_df <- data.frame(plasma_time_df)

#add genes
fcm_plotting_df$Feature <- row.names(fcm_plotting_df)

#bind cluster assinment
fcm_plotting_df$cluster <- mfuzz_cl$cluster

#fetch the membership for each gene/top scoring cluster
fcm_plotting_df$membership <- sapply(1:length(fcm_plotting_df$cluster),function(row){
  clust <- fcm_plotting_df$cluster[row]
  mfuzz_cl$membership[row,clust]
})


#subset the dataframe by the cluster and get it into long form
#using a little tidyr action
cluster_plot_df <- fcm_plotting_df %>%
  dplyr::select(.,1:length(colnames(fcm_centroids)), membership, Feature, cluster) %>%
  tidyr::gather(.,"sample",'value',1:length(colnames(fcm_centroids)))

#order the dataframe by score
cluster_plot_df <- cluster_plot_df[order(cluster_plot_df$membership),]
#set the order by setting the factors using forcats
cluster_plot_df$Feature = forcats::fct_inorder(cluster_plot_df$Feature)

#subset the cores by cluster
ggplot(cluster_plot_df, aes(x=sample,y=value)) + 
    facet_wrap(~cluster, scales="free_y") +
    geom_line(aes(colour=as.factor(cluster), alpha=membership-0.1, group = membership), lwd=1) +
    #scale_colour_gradientn(colours=c('blue1', 'red2')) +
    #scale_colour_gradient(low = "white", high = "red") +
    #scale_color_viridis(option = "mako", direction = -1) + 
    #this adds the core 
    geom_line(data=centroids_long, aes(sample,value, group=cluster), lwd=1.5, color="black",inherit.aes=FALSE) +
    xlab("Time") +
    ylab("Expression") +
    labs(title= paste0("Per-cluster membership"),color = "Score") + 
    theme_cowplot()

```


```{r}

#start with the input data
fcm_plotting_df <- data.frame(plasma_time_df)

#add genes
fcm_plotting_df$Feature <- row.names(fcm_plotting_df)

#bind cluster assinment
fcm_plotting_df$cluster <- mfuzz_cl$cluster

#fetch the membership for each gene/top scoring cluster
fcm_plotting_df$membership <- sapply(1:length(fcm_plotting_df$cluster),function(row){
  clust <- fcm_plotting_df$cluster[row]
  mfuzz_cl$membership[row,clust]
})


#subset the dataframe by the cluster and get it into long form
#using a little tidyr action
cluster_plot_df <- fcm_plotting_df %>%
  dplyr::select(.,1:length(colnames(fcm_centroids)), membership, Feature, cluster) %>%
  tidyr::gather(.,"sample",'value',1:length(colnames(fcm_centroids)))

#order the dataframe by score
cluster_plot_df <- cluster_plot_df[order(cluster_plot_df$membership),]
#set the order by setting the factors using forcats
cluster_plot_df$Feature = forcats::fct_inorder(cluster_plot_df$Feature)

#subset the cores by cluster
ggplot(cluster_plot_df, aes(x=sample,y=value)) + 
    facet_wrap(~cluster, scales="free_y") +
    geom_line(aes(colour=membership, group=Feature), lwd=1, alpha=0.5) +
    #scale_colour_gradientn(colours=c('blue1', 'red2')) +
    scale_colour_gradient(low = "white", high = "red") +
    #scale_color_viridis(option = "mako", direction = -1) + 
    #this adds the core 
    geom_line(data=centroids_long, aes(sample,value, group=cluster), lwd=1.5, color="black",inherit.aes=FALSE) +
    xlab("Time") +
    ylab("Expression") +
    labs(title= paste0("Per-cluster membership"),color = "Score") + 
    theme_cowplot()

```



```{r}

#start with the input data
fcm_plotting_df <- data.frame(plasma_time_df)

#add genes
fcm_plotting_df$Feature <- row.names(fcm_plotting_df)

#bind cluster assinment
fcm_plotting_df$cluster <- mfuzz_cl$cluster

#fetch the membership for each gene/top scoring cluster
fcm_plotting_df$membership <- sapply(1:length(fcm_plotting_df$cluster),function(row){
  clust <- fcm_plotting_df$cluster[row]
  mfuzz_cl$membership[row,clust]
})

k_to_plot = 3

#subset the dataframe by the cluster and get it into long form
#using a little tidyr action
cluster_plot_df <- dplyr::filter(fcm_plotting_df, cluster == k_to_plot) %>%
  dplyr::select(.,1:length(colnames(fcm_centroids)),membership,Feature) %>%
  tidyr::gather(.,"sample",'value',1:length(colnames(fcm_centroids)))

#order the dataframe by score
cluster_plot_df <- cluster_plot_df[order(cluster_plot_df$membership),]
#set the order by setting the factors using forcats
cluster_plot_df$Feature = forcats::fct_inorder(cluster_plot_df$Feature)

#subset the cores by cluster
core <- dplyr::filter(centroids_long, cluster == k_to_plot)

ggplot(cluster_plot_df, aes(x=sample,y=value)) + 
    geom_line(aes(colour=membership, group=Feature), lwd=1, alpha=0.5) +
    #scale_colour_gradientn(colours=c('blue1', 'red2')) +
    scale_colour_gradient(low = "white", high = "red") +
    #scale_color_viridis(option = "mako", direction = -1) + 
    #this adds the core 
    geom_line(data=core, aes(sample,value, group=cluster), lwd=1.5, color="black",inherit.aes=FALSE) +
    xlab("Time") +
    ylab("Expression") +
    labs(title= paste0("Cluster ",k_to_plot," Expression by Time"),color = "Score") + 
    theme_cowplot()

```

Extract membership scores for each gene by cluster:

```{r}

# Extracts membership values 
acore <- acore(expData, cl, min.acore=0)


# Create a combined dataframe of the combined clusters
acore_list <- do.call(rbind, lapply(seq_along(acore), function(i){ data.frame(CLUSTER=i, acore[[i]])}))

```


https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7893783/ - eca-miR-186
https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7331607/ - F7C1Y3
	
eca-miR-126-5p


# dtwclust

```{r, fig.show='hide', message=FALSE}

pc <- tsclust(na.omit(plasma_time_df), type = "partitional", k = 6L, 
              distance = "dtw", centroid = "pam", 
              seed = 3247L, trace = TRUE,
              args = tsclust_args(dist = list(window.size = 20L)))



time <- gsub("Day_", "", colnames(plasma_time_df))

pc_plot <- plot(pc, time=time) + 
            theme_cowplot() +
            xlab("Time (days") +
            ylab("Z-score") 

```

```{r, echo=FALSE}

pc_plot 

```

```{r, fig.show='hide', message=FALSE}

fc <- tsclust(na.omit(plasma_time_df), type = "fuzzy", k = 6L,
              distance = "L2",
              seed = 123L)


fc_plot <- plot(fc, time=time) + 
            theme_cowplot() +
            xlab("Time (days") +
            ylab("Z-score") 


```

```{r, echo=FALSE}

fc_plot 

```

https://stackoverflow.com/questions/42197984/how-to-get-the-result-of-dtwclust
